\chapter{Introduction}
\label{chap:introduction}
\tightlists

As a geneticist, my main goal is to shed more light on how genotype results in phenotype. These can be phenotypes at any level, for example at the cellular level, tissue level or organismic level. We know that DNA sets the blueprint for all the phenotypes of an organism. The first and arguably one of the most important steps in the process of going from DNA to cellular machines is gene expression. DNA gets transcribed to RNA and these are then translated to proteins, and these are the molecules that set up life as we know it - both inside us and around ourself. We function as who we are due to the concerted action of these DNA, proteins and RNA.

So, to understand how genotype translates to phenotypes it is key to understand gene expression. There are indeed a large number of research programs devoted to understanding gene expression. This thesis also attempts to answer a question related to gene expression. Specifically, how is gene expression regulated amongst a population of cells?

Let's peek inside a single-cell. In most cell-types in a human body there are two copies of each chromosome. A small part, about 1\% of this DNA is what codes for proteins \cite {lander2001n}. Through a series of molecular steps that we don't quite yet understand, genes are turned on and off in a precise manner during development. It is this precise turning on and off of genes that generates a whole organism from a single-cell and the multitude of cell-types and organisms inside and around us. When we hear of such precision, one might think of a well oiled machine that is entirely predictable. A machine that turns on and off in an entirely predictable manner within a cell. But this is not what we observe.

We have become really good at measuring gene expression both in a population of cells \cite{griffith2015pcb}, and within single-cells \cite{macosko2015c, newman2006na, xia2019pnasu, raj2006pb}. If it were true that the gene expression operates in a predictable manner, if we start with a single cell and follow gene expression in it's daughter cells, the cells would express identical amounts of RNA and protein. However this is not what one observes \cite{elowitz_stochastic_2002}. There is a tremendous amount of cell-to-cell variability in gene expression amongst a population of cells. This thesis deals with understanding this cell-to-cell variability - how does this cell-to-cell variability come about i.e its sources, and does this variability have any phenotypic impact i.e its consequences. In this thesis, I use cell-to-cell variability in gene expression and gene expression noise interchangeably.

\section{Gene expression variability exists amongst cells}

Gene expression is the result of probabilistic interactions between molecules in the cell. These molecules include the two copies of DNA, transcription factors, RNA polymerase amongst others. The low numbers of molecules involved in transcription can results in  variable gene expression across cells and time. It is still a mystery how even in the presence of these fluctuations  biological systems function in a reliable manner.

The phenomenon of variability between cells is not new. Novick and Weiner \cite{novick_enzyme_1957} \cite{raj_nature_2008} described a phenonmenon in 1957 where given a population of E. Coli, the fraction of cells that produce beta galactidase is not deterministic at low inducer concentrations. This fraction varies with the concentration of the inducer. The increase in the concentration of inducer only increases the chance that a bacterium produces beta galactidase. Given a fixed concentration of the inducer, it is not possible to predict with certainty if a single bacterium will produce beta galactidase but it is possible to estimate the probability of this event. This is fascinating. This paper is also one of the earliest works that I am familiar with that talks about variable cell-state - there's an induced state and a non-induced state and the whole question is about the transition from one cell-state to another. The authors also compare the enzyme induction state to a mutation.

Now let us turn to variability in gene expression. In 2002, Elowitz et al.  \cite{elowitz_stochastic_2002} showed using two-colored reporter genes in E. Coli that identical genes on two different chromosomes show cell-to-cell variability in expression. The authors decomposed this variability into two components. The first is intrinsic noise that is specific to a locus. Intrinsic noise is due to the local interactions of molecules involved in transcription at a locus. These might include promoters switching between active and inactive states, proteins binding to enhancers, recruitment of RNA polymerase and so on. The second source of variability is extrinsic noise. Extrinsic noise is due to differences between cells in what is commonly termed the cellular milieu. The cellular milieu can include upstream regulators of transcription such as transcription factor concentrations, number of ribosomes in the cell, differences in cell cycle stage, differences in mitochondrial number etc. These differences in the cellular environment is likely to affect the expression of multiple genes. Several other studies that came out around roughly the same confirmed the observation that genetically identical cells in the same environment display variability in gene expression, this phenomenon is observed in cells from multiple organisms. \cite{raser_control_2004,blake_phenotypic_2006,blake_noise_2003,raser_noise_2005,volfson_origins_2006}.

Following these studies variability in gene expression has also been observed and modeled at the RNA level. As an example of an early study, in 2006 Raj et al \cite{raj2006pb} counted the number of mRNA molecules of using fluorescence in situ hybridization (FISH). They found a large variation in the number of mRNA molecules between cells of the same cell type. They find that most of the variability they measure is due to intrinsci noise. They then modeled the mRNA distributions using a stochastic model of gene transcription and show transitions between an inactive state and a state of active transcription can result in such cell-to-cell variability. They observe variations in burst size, which they attribute to transcriptional activators, but they did not find that the data fit changes to burst frequency. This work marked the beginning of a series of studies that used the cell-to-cell variability in the mRNA molecules to understand mechanisms of transcription \cite{symmons2016mc} \cite{chen2018tissues}.

More recently it has become possible to visualize transcription in live cells and infer mechanisms of transcription. For example, in 2019 Rodriguez et al. \cite{rodriguez_intrinsic_2019} used live-cell RNA imaging to look at the regulation of hte estrogen-responsive gene TFF1. The authors observe periods where the gene is transcriptionally active or inactive. The inactive periods vary hugely, they can range from periods in minutes to many hours. The active periods are much  shorter and are of the order of 16 minutes on average. They also find instances of allelic coupling where transcription of one allele influences the transcription of the other allele more likely. Deletion of an enhancer region increases the inactive time period, which fits with the literature reflecting that enhancers control burst frequency. Using single-cell RNA sequencing data, the authors postulate that the deep repressive state that is observed for TFF1 could extend to other genes in certain functional categories such as secretion, transmembrane and signal peptides. Their modeling reveals a deep repressive state  which a gene could spend hours in a repressive state. It remains to be seen if this deep repressive state is observed in cells in the future, this paper used observation times of 14 hours which is insufficient to observe this state. If this state is real it would be fascinating to understand the mechanisms at the DNA level that underlie this state again revealing the fascinating ways in which cell-to-cell variability can unveil mechanisms of gene expression.

As time passed, technical innovations have helped us peer at different molecules within cells and they have all revealed that cell-to-cell variability in gene expression is ubiquitous. But does this cell-to-cell variability matter?

\section{Cell-to-Cell variability can result in functional differences}

An increase in cell-to-cell variability in gene expression has been observed in cells undergoing differentiation. Richard et al.  \cite{richard_single-cell-based_2016} looked at a well-studied differentiation system of chicken erythroid progenitors differentiating into erythrocytes. The authors showed that chicken erythroid cells show an increase in variability of gene expression of relevant genes prior to differentiation. Similar results were obtained by Mojtahedi et al  \cite{mojtahedi_cell_2016} while studying the differentiation of murine heamatopoetic progenitor cells into myeloid or erythroid cells. The authors hypothesize that this surge in variability allows the cells to transition out of stable cell-states and transition to different cell-states. 

Other examples have shown that genetically identical cells don't all reprogram at the same rate. Hanna and Saha et al. showed that reprogramming of somatic cells to iPSCs is stochastic with a random subset of initial cells completing reprogramming  \cite{hanna_direct_2009}. The authors showed that eventually all somatic cells reprogram to pluripotent stem cells when induced with transcription factors but the rate at which cells reprogram can be increased by increasing the cell-division rate. Similar stochasticity has been observed when fibroblasts were reprogrammed into induced endoderm progenitor cells \cite{biddy_single-cell_2018}.

Variability in phenotype has also been shown to be functionally important at the organismal level. In C. elegans, two identical cells from different lineages known as alpha cells always give rise to exactly one ventral uterine progenitor cell and one anchor cell \cite{seydoux_cell_1989}. Both the alpha cells have the potential to generate either of the two downstream cell-types but the cell fate is decided in a stochastic and co-ordinated manner. This case is a well-studied model of a stochastic cell-fate decision where a robust outcome is produced as a result of a stochastic choice. In another example in C. elegans, mutations that cause incomplete penetrance have been shown to be due to increased variability in the expression of cell-fate determining genes  \cite{raj_variability_2010}.  The mutants display increased variability in the expression of a gene called *end-1*. Cells with high-levels of *end-1* are able to activate downstream targets in the intestinal fate specification network whereas the cells with low levels of *end-1* do not. 

In bacteria, individual bacterium often display a persistor phenotype and grow at slower rates by expressing different subsets of genes. This confers the bacteria an ability to survive adverse changes such as antibiotic treatment that might otherwise terminate a homogenous population \cite{veening_bet-hedging_2008}. In yeast, variability has been observed in the utilization of the sugar galactose. The Gal genes that are responsible for the metabolism of galactose pathway display a bimodal expression pattern \cite{acar_enhancement_2005}. This bimodality has been explained using stochastic modelling of the expression of genes in this pathway. These results all show how biological variability, arising from variability in gene expression, can have functional consequences. 


\section{RNA sequencing can be used to uncover cell-to-cell variability}

RNA sequencing, both at the bulk and single-cell level, can be used as a readout of functional cell-states. A prominent example is the work by Chang et al. \cite{chang_transcriptome-wide_2008}. Chang et al. isolated mouse heamatopoetic progenitors with different levels of a haematopoetic stem-cell marker Sca-1.  Sca1 expression showed reproducible variability across cells that was not just due to technical error in measurement. The authors used microarrays to show that cells with low amounts of Sca-1 had different transcriptomes compared to cells with low amounts of Sca-1. Furthermore, cells with higher levels of Sca-1 have an increased likelihood of differentiation towards the Myeloid lineage while cells with low amounts of Sca-1 are more prone to differentiate towards the Erythroid lineage. Similar examples have been observed for embryonic mouse cells with different amounts of a stem-cell marker Nanog \cite{kalmar_regulated_2009}. The authors showed that cells with low levels of Nanog exhibit increased variability in gene expression and are prone to differentiate.

Computational modeling approaches are important to understand expression differences between single-cells. The single-cell RNA seq data gives us a static snapshot of a dynamic process. Every cell on a PCA plot is a summary of an enormous number of events. These cells are transitioning through different cell-types in a differentiating system or through different cell-states within a homogenous cell-type population. A commonly used approach to understand this dynamical process is to cluster cells with similar transcriptomes together. The predicted paths of cellular trajectories are inferred by drawing a principal curve through the cells. By correlating the expression patterns along the trajectory with known biology we can infer the processes active in these cells. \cite{trapnell_dynamics_2014}. Another recently developed method called RNA-velocity allows us to infer the near-term future transcriptome of cells. This is done by fitting a kinetic model to the ratio of spliced to unspliced transcripts of each gene. This allows the number of inferred transcript counts of different genes to be estimated. The inferred transcriptome is then plotted on a PCA or tSNE plot to infer the position where a given cell's transcriptome is headed.  \cite{manno_rna_2018} Cell state transitions can be modeled as Markov processes \cite{stumpf_stem_2017} and the RNA-velocity approach can be used to infer transition rates of the model. Using such computational models can help answer interesting questions. For example, we can try to understand the proportion of time cells spend in a certain cell-state and transition to different cell-states, how many cell-states explain the observed data  \cite{chang_transcriptome-wide_2008}, what are the characteristic markers of different cell-states, and what are the genes are driving cell-state transitions \cite{furchtgott_discovering_2017}.

A recent study used single-cell RNA-seqeuencing to estimate the burst frequency and burst size of about 7000 genes genome-wide \cite{rodriguez_intrinsic_2019}. This study used Smart-seq2 technology which provides coverage across the transcript and shows a reduced 3' bias compared to alternative technologies such as 10X. \cite{picelli2013nm} The authors estimated the average waiting time for a transcriptional burst of a gene is 4 hours though a smaller fraction of genes had waiting times of greater than 20 hours for a burst. Smaller genes have higher burst size, and the authors suggest this is expected for housekeeping genes which are usually compact and consistently expressed in most tissues. Genes with TATA elements also have a higher burst size. By applying their method to a different cell types the authors show that many DE genes show a difference in burst frequency between cell-types. The difference in burst frequency also correlates with histone marks that mark enhancer activity. The authors conclude that core promoter elements play a major role in determining burst size and enhancer elements determine burst frequencies and that cell-type specific gene expression is determined by changes in burst frequencies. This sort of genome-wide study of transcriptional kinetics can be enabled by single-cell RNA sequencing and provide a supplement to more focused, less noisy assays looking at transcription of a single reporter gene. As the capture technology for scRNA-seq improves we will get better estimates of genome-wide transcriptional kinetics and how they change across different dimensions, for example with genetic background or with age or with disease.


\section{ Noise can  be biological or technical }

\section{Genetic control of noise}

%figure1
\begin{figure}[t!]  
    \centering
    \includegraphics[width=\linewidth, scale=0.25]{figures/intro/intro_outlier1.png}
    \caption[Distribution of a Transcription Factor across cells.]{%
        \textbf{Distribution of a Transcription Factor across cells}
        The protein levels of any given transcription factor is a distribution across cells. Some cells have high amounts of the protein, some have intermediate levels, and some cells have lower amounts of the protein.
    }
    \label{fig:intro1}
\end{figure}

An important question that still remains to be answered is are the mean and noise controlled by different genetic regulatory elements.
We have made a lot of progress in identifying loci that are associated with the mean level of expression \cite{gtex_consortium_genetic_2017} \cite{vosa2021ng} \cite{wen2017pg}. These studies have identified eQTLs for almost every gene in the genome and about half of all GWAS hits can now be mapped to eQTLs \cite{gtex_consortium_genetic_2017}. However this only looks at the mean level of expression. Could a similar approach be used to find loci that find variance QTLs?

Studies have looked explicitly for variance QTLs. Sarkar et al. tried to find variance QTLs using the well characterized YRI cell lines(n = 53) from the HapMap project \cite{gibbs2003n}. Briefly, they collected single-cell RNA-sequencing data from N individuals using 10X. Then they fit a negative binomial to each gene and came up with a noise measure for each gene. Since they also had genotype information for each of these individuals they looked for variance QTLs. What di they find? They found all variance QTLs were also mean QTLs. THey concluded that their sample size was too small and that they would need around 4000 number of samples to find the strongest mean independent variance QTLs at 80\% power. This study provided some methodological innovations despite being a negative result. There has been one other study that has used a simlar approach to find variance QTLs. Francesca et al. used 10X scRNA-seq data from X sampels and attempted to find QTLs. They found...

Prior to the publication of the study by Sarkar et al. I also looked at the single-cell data from their group through a collaboration with their group. I looked at cell-to-cell variability using different approaches. My variance QTL analysis also detected variance QTLs that were explained entirely by mean QTLs. I also looked at every gene's expression for every individual, and tried to find genes whose variance in gene expression was disproportionate to their mean levels. I found a set of genes that had variance above expectation consistently in many individuals (shown in red in \fref{fig:intro1}). Upon closer inspection, many of the genes that displayed this property were enriched for Mitochondrial genes. We hypothesized that mitochondrial genes might have increased variability for some functional reason. However, it is a known observation in single-cell RNA sequencing that cells that are unhealthy or dying exhibit higher levels of mitochondrial transcripts \cite {stuart2019c}, hence we did not follow up on this observation. We also used this data to look pathway specific gnees that displayed abnormal levels of variability. However, I did not find any pathways that displayed this property. Like Sarkar et al. I attribute this partly to higher technical noise of single-cell RNA sequencing as well as lower power to detect real effects due to the relatively small sample size.

\section{Selection on noise levels}

A long-standing question is whether the cell-to-cell variability of different genes are under natural selection, or does selection act only at the mean level. The alternate hypothesis is that the noise is just unconstrained by selection and all the selection acts on the mean level of expression. As discussed in the previous section, we are now beginning to identify regulatory elements that are associated with noise but not mean levels. If these regulatory elements are under selection, or if they are conserved across species, these could indicate a possible important functional role for noise that's independent of the mean level of expression of that gene. Some studies have explored this question.

Newman et al. \cite{newman2006na} measured the noise level for 2500 genes in the the yeast genome under rich and minimal media conditions. They did this by GFP-tagging each gene and measuring the fluorescence level on a cytometer. This collection of GFP-tagged yeast strains has been an invaluable resource, and indeed I used a couple of strains from this project for my early exploratory noise projects that I described in this chapter. Newman et al. found that proteins that respond to environmental cues are more noisy than other groups of genes like housekeeping genes. Zhang et al. \cite{zhang2009msb} confirm this observation for plasma transporter genes and use mathematical models to argue that higher noise promotes evolvability of favourable gene expression levels, and could be under positive selection. 

Fraser et al. \cite{fraser2004pb} looked at noise in two classes of genes in yeast - essential genes, and genes involved in protein complex subunits and compared the noise levels in these two groups to the rest of the genes in the genome. They estimated protein production rates, and transcription rates for every gene in the genome and found that the essential genes and the genes that are part of protein complexes display a lower rate of translation, and a high rate of transcription and decay - a strategy that minimizes noise. The authors speculate that this strategy might have been an evolutionary mechanism to minimize noise in genes that are essential to life.

Batada et al. \cite{batada2007ng} asked whether genes that are essential are in regions are in regions of the genome that have low nucleosome occupancy, and whether noise-sensitive genes cluster with essential genes in the genome. They reason that genes that are essential might have evolved to be in regions of the genome that are less noisy. Using Myc-tagged histone data they indeed find an anti-corrrelation between regions of high essential gene density and nucleosome occupancy. They find a dearth of essential genes in subtelomeric regions which are high noise regions of the yeast genome.

Metzger et al. \cite{metzger2015n} have looked at the promoter of a yeast enzyme that is involved in glucose metabolism in multiple yeast species. They looked at both polymorphisms that have fixed after undergoing selection through time and new mutations, induced through mutagenesis, which have not yet undergone seleection. They find that the polymorphisms have a slightly smaller effect size on the noise level compared to new mutations, suggesting the action of selection in removing some of the noise affecting polymorphisms in the TDH3 promoter. Naturally occuring haplotypes show lower amounts noise than what would be expected under neutral evolution.

The studies cited in this section have looked for signatures of selection acting specifically on the noise level. However for some of these studies it has been hard to attribute this signal entirely to the variability, and not to the mean level of expression. Indeed, most of these studies don't rule out alternative explanations for the signals of selection that are observed in the discussion sections.

\section{Noise in transcription factors}

%figure1
\begin{figure}[t!]  
    \centering
    \includegraphics[width=\linewidth, scale=0.25]{figures/intro/intro_gfp_density_X.pdf}
    \caption[Distribution of Transcription Factor across cells.]{%
        \textbf{Distribution of Transcription Factor across cells}
        caption goes here
    }
    \label{fig:intro2}
\end{figure}

Consider the distribution in expression of a single transcription factor protein as shown in figure \fref{fig:intro2}. Transcription factors play vital roles in development and disease, and can convert between cell-types. \cite{weirauch2011ahotf} \cite{levine2003n}. Could it then be that a cell that expresses high amounts of a transcription factor behave differently from another cell with an identical genome and growing in essentially identical environment but expressing lower amounts of the same transcription factor? Indeed, several studies have shown this to be the case across different organisms.

Wernet et al. \cite{wernet_stochastic_2006} examined the cell-fate choice in the Drosophila photorecptor cells. The drosophila eye consists of optical units called ommatidia which choose between two subtypes pale and yellow which is determined by the pairs of rhodopsin genes expressed in them. How this fate is chosen amongst genetically identical cells had been a fascinating question. The authors of this paper showed that the expression of a single transcription factor spineless is necessary and sufficient to specify the mosaic cell-fate in the Drosophila eye. Spineless is stochastically expressed in a majority of the cells which then adopt the yellow subtype. The knockout of Spineless results in all of the ommatidia adopting the blue subtype. Overexpressing spineless results in all of the ommatidia adopting the yellow subtype. This fascinating and crucial development process is thus controlled by the stochastic expression of a single transcription factor.

In \emph{C. elegans} somatic primordium four cell have the potential to be the Anchor cell (AC) which is an important cell in the gonads. \cite{attner2019cb}. How only one of these four cells becomes the AC is determined by the the transcription factor HLH-2. The rest of the four cells become the ventral uterine precursor cells (VU) cells. The relative timing of the expression transcription factor influences cell-fate in this system. The parent that expresses HLH-2 first results in the daughter becoming the VU cell. This onset of expression is also linked to the birth order of the parent cells, the parent cell that is born first also expresses HLH-2 first and adopts the VU fate. However when the birth order difference is  relatively short, the AC/VU decision appears to be random. This system is different from the previous study since it is stochastic while also being deterministic in that there is always only one AC cell, and this is determined by the birth order of the ancestor cells. The authors speculate that other stochastic outcomes might be resolved by finding other such deterministic elements. 

In Bacteria, Maamar et al. \cite{maamar2007s} examined the emergence of the transient state of competence that enables a bacterium to uptake DNA from the environment. They showed that the noise in comK the master regulator of the competence pathway involving autorogelation. This autoregulation results in bistability where in one state comK expression is low, and in the other state the comK expression is higher than a critical threshold and enables the bacterium to become competent. They show that the noise in comK is mostly intrinsic.

In this section we have seen several examples across species how the cell-to-cell variability of a single transcription factor can result in important functional differences between cells. An important follow-up question that emerges from this observation is what determines the noise levels of a certain gene? While the actual regulatory mechanisms can vary a lot between genes, certain general mechanisms of transcription can be examined for their impact on noise.

\section{Sources of noise}

\subsection{Effect of genomic features and chromatin on noise}

Several studies have looked at the effect of genomic location by integrating the same reporter gene in multiple locations. These sutudies used sequencing, time-lapse microscopy and flow cytometry approaches to look at bulk RNA and single-cell protein levels in cells. Our approach in Chapter 3 uses a similar integration strategy to look at RNA levels in single-cells to uncover the effects of cellular and genomic environment on gene expression noise.

Akhtar et al. \cite{akhtarw_vansteenselb:ChromatinPosition2013} integrated the same reporter gene in roughly 30000 locations along the mouse genome in ES cells by devising a barcode based technology called Thousands of Reporters in Parallel (TRIP). They measured the mean level of RNA expression of the reporter by using bulk RNA sequencing. The expression levels of the reporter genes vary more than 1000-fold in these locations. Nearby locations are correlated in terms of their expression and the authors used a HMM to separate the genome into permissive and non permissive domains. The non-permissive domains overlap with lamina associated domains and late-replicating domains. Permissive IR domains coincide with gene-dense and active regions. The integration locations that are promixal to genes are 10-fold more active than those located far from genes. Our technology in Chapter 3 is highly inspired by the TRIP method of Akhtar et al. The first part of our experimental pipeline is derived from the protocols of Akhtar et al. We wanted to extend TRIP to look at the factors underlying the variance of a reporter gene and not just the mean.

Dar et al. \cite{dar2012pnas} did a genome-wide study of burst frequency and burst size by using time-lapse fluorescence mircoscopy at 8,000 loci. The authors use autocorrelation time of the fluctuations as a third axis, apart from mean and noise, to understand the dynamics at different loci. The third dimension allows the authors to constrain the two-state model. They used HIV-1 lentivirus to integrate GFP driven by three different promoters. They find that bursty transcription predominates over consitutive transcription genome-wide. Both burst size and burst frequency vary in equal measure across the genome. Weaker loci modulate the burst frequency to increase expression, whereas after a certain level of expression is reached burst size is modulated to increase expression. They perturbed transcription at this loci using histone deacetylase inhibutors. They find that both burst frequency and burst size increase with increasing expression. 

Dey et al. \cite{dey2015msb} used a lentiviral approach to integrate reporter genes at hundreds of promoters. Their approach used a dual reporter system to quantify the GFP reporter by flow-cytometry and the RNA levels by smFISH. They also assayed chromatin accessibility at the integration sites and found that higher noise locations are enriched for inaccessible chromatin. They also fit a two-state stochastic model to the data and found that the burst size determines the RNA mean while the promoter On rate influences the noise. This approach by Dey et al. provided a lot of novel insight, however multiple clones had to be measured individually as a result chromatin analysis was done on a small set of locations and the locations used in this study were not mapped. A high-throughput approach that allows for potential mapping of thousands of locations would provide more fine-grained insights.

While all three prior approaches described above furthered our understanding of how mean RNA expression and noise at the protein level changed with genomic locations, there are still some unanswered questions. Can we understand the noise at the RNA level in more locations? How does the cell-state of a cell influence the gene expression noise?

\subsection{Effect of cell state on phenotypic variability}

The previous section highlighted the effect of chromatin and genomic features of the location where a gene is situated on expression noise. A complementary factor affecting gene expression noise is the cell-state. Noise can be decomposed into intrinsic and extrinsic components \cite{swain2002pnas} \cite{raser_noise_2005}. The intrinsic component is related to the local transcriptional machinery at the gene. The extrinsic component of noise is at the cellular level and varies from cell-to-cell - for example differences in cell-cycle, cell size etc. A part of this extrinsic component can be captured by measuring the transcriptomes of individual cells \cite{macosko2015c}. This set of transcriptional measurements of each cell is also referred to as the cell-state of the cell. Recent studies have begun to examine how cell-state affects gene expression noise.

Foreman et al. \cite{foreman_mammalian_2019} measured the role of cell-state in explaining the amount of variance related to Calcium signaling after ATP stimulation. The authors also used multiplexed error robust RNA fluorescent in situ hybridization (MERFISH) to measure the cell-state of each cell by measuring the mRNA levels of 150 genes mostly related to Calcium signaling and some genes related to the cell cycle. The question that the authors were examining is how much of the variance in the Calcium signaling response can be explained by global factors such as the cell-state, cell-cycle, cell-volume and differentiation state. They also set to examine why most gene expression distributions are overdispersed. The answer is that most of the variability can be explained by cell-to-cell differences such as the cell-state and cell-size. Whatever variability remained after accounting for the cell-to-cell differences was Poissonian. This result argues that most of the noise is not gene-specific and that the shared noise dominates the allele-specific noise. This paper offers an intriguing conclusion that transcriptional bursting might not be as prevalent as previously described and that after accounting for the cell-state most genes don't follow a bursty model of transcription. While it remains to be seen how much of this finding translates to other models of study outside Calcium response, it becomes necessary to measure cell-state in any study related to gene expression noise. Our method in Chapter 3 meets this need.

Torre et al. \cite{torre2021ng} used a genetic screening approach to find genes that regulate a drug-resistant cell-state. This drug-resistant cell-state is initially transient and rare (about 1 in 2000-3000 cells) but upon treatment, these become a stable cell-state.  The authors were interested in understanding what are the pathways that control the transient drug resistant state in the first place. This approach could potentially uncover genes that are missed in a classic drug-resistance screen. They tested two genes identified in the screen DOT1L and LATS2 which increased the frequency of the primed state and BRD2 which resulted in fewer primed cells. Along these lines, they found that the knockout of DOT1L and LATS2 results in larger tumor volumes and knockout of BRD2 results in smaller tumor volumes. This sort of genetic-screening approach can help identify the regulators of cell-states that underlie phenotypic variability. One thing that is essential to know is some sort of marker of the cell-state one is interested in. In the Torre et al. study they could sort cells based on expression of NGFR and EGFR which mark the primed cells.

Topoleweski et al. \cite{topolewski2022ssb} looked at cell-to-cell heterogeneity in the JAK-STAT signaling pathway response. The authors wanted to test if most of the variability could be explained by differences between cells in their cytoplasmic content, which would reflect cell-state or if the variability could be explained by just the local intrinsic fluctuations of genes also referred to as intrinsic noise. The authors used an interesting approach by fusing together fibroblast cells to form binuclear synctia. The reason that the fused cells could approximate as two cells with the same molecular content. The authors find that the response of two nuclei is almost always identical and explains 91\% of the variance. The remaining noise is just around 9\%. The authors speculate that the cell-state or the molecular phenotype of a cell coudl result in the cells interpreting the same signal differently. Thus the phenotypic variability doesn't argue against reliable functioning of single-cells but reflects variability in the content of cells. It still remains to be well understood how the differnt molecular phenotypes can result in variable outcomes, can we for example predict how a cell will respond to a stimulus using its transcriptome. Such approaches are likely to be possible in the future.

\subsection{ How does noise propagate within a network}

Extrinsic noise reflects global cell-to-cell differences such as cell-cycle, cell-size and affects many genes within a cell. It is still unclear what are the different levels at which extrinsic noise can act. For example can extrinsic noise act at the level of pathways?

One of the questions that I asked early in my thesis work was whether the noise in an upstream regulator of a pathway propagates within it's network. The way I tested this was by using one of the GFP-tagged yeast strain for the gene TDH2 from Newman et al.\cite{newman2006na}. TDH2 is an enzyme that plays a role in the glycolysis pathway. We hypothesized that levels of this enzyme in a given yeast might manifest as differences in the transcriptomes between yeast and possibly phenotypic differences between yeast. If the noise in TDH2 propagated downstream to the other genes in the pathway then we would be able to see differences in the target genes. 

\begin{figure}[t!]  
    \centering
    \includegraphics[width=\linewidth]{figures/intro/intro_tdh2_facs.png}
    \caption[Representative sort results of TDH2]{%
        \textbf{Representative sort results of TDH2}
        Yeast with GFP-tagged copies of the gene TDH2 were grown out and sorted on a flow sorter. Three populations of yeast with relative high, medium and low amounts of TDH2-GFP were sorted into individual tubes.
    }
    \label{fig:intro3}
\end{figure}

I sorted out three populations of cells as shown in \fref{fig:intro3}. One population expressed TDH3 at a relatively high level, marked by high levels of GFP, another population at a lower level, and another population at an intermediate level between the high and low levels.  I sorted these two populations into three separate vials, extracted RNA from the three populations and measured the transcriptomes using RNA-seq.

\begin{figure}[t!]  
    \centering
    \phantomlabel{fig:intro4a}
    \phantomlabel{fig:intro4b}
    \phantomlabel{fig:intro4c}
    \includegraphics[width=\linewidth, scale=0.5]{figures/intro/intro_clustering_expectedresults.pdf}
    \caption[Expected results from sorting out populations with high, low and medium levels of a protein as in \fref{fig:intro2}]{%
        \textbf{Expected results from sorting out populations with high, low and medium levels of a protein as in \fref{fig:intro2}}
        A) The transcriptomes don't show any discernible pattern. B) The transcriptomes separate out by the levels of the sorted protein. C) The transcriptomes separate out by a technical factor like batch.
    }
    \label{fig:intro4}
\end{figure}

What might we expect from the transcriptomes obtained from an experiment. There are three posibble outcomes if we use a clustering analysis like PCA. If the levels of the enzyme did not result in any distinct transcriptional changes, and by extension phenotypic differences amongst populations, then we would expect the transcriptomes from the high, low and medium populations to overlap as seen in Figure \fref{fig:intro4a}. If the high and low levels of the enzyme result in distinct transcriptional states, due to upregulation of the pathway in the high cells, we would expect to see the results in Figure \fref{fig:intro4b} where the high, low and medium levels of TDH2 result in distinct transcriptional states that cluster separately. This is similar to what Chang et al see in embryonic stem cells with different levels of Sca1 \cite{chang2008n}. If the experiment-to-experiment variability and batch effects of the experiment dominate the within day variability then we would expect to see the results in Figure \fref{fig:intro4c} where the samples cluster by replicate i.e the day of the experiment. This framework helps us look at the observed data.

\begin{figure}[t!]  
    \centering
    \includegraphics[width=\linewidth, scale=0.5]{figures/intro/intro_tdh2_clustering_timepoints.pdf}
    \caption[Transcriptomes of yeast don't separate out by TDH2 levels]{%
     	\textbf{Transcriptomes of yeast don't separate out by TDH2 levels}
     	We sorted out populations of yeast with high, low and medium levels of TDH2 (3 technical replicates each) as shown in \fref{fig:intro3}. We then extracted RNA and performed RNA-seq. We used PCA to visualize the transcriptomes and find that they don't separate out by the sorted TDH2 levels.
    }
    \label{fig:intro5}
\end{figure}

We counted the transcripts for every expressed gene and performed a PCA analysis on the tnormalized transcript counts. These results are seen in \fref{fig:intro5} The transcriptomes separated out by the day of experiment along PC1. Furthermore, when we looked at individual transcript changes between high and low populations there were very few differentially expressed genes. This result looks like our expected result in \fref{fig:intro4c}. The variability between replicates is higher than the variability within a replicate. Since there were so few differentially expressed genes between teh high and low populations we didn't find any pathway related differences between these populations as well.

While this early experiment of mine was largely a negative result I was still able to get started and thinking about transcriptional variability and its possible functional effects. This experiment was technically challenging since large number of yeast had to be sorted in order to obtain enough RNA for library construction. We had to ensure that the yeast were processed in such a way as to not allow for transcriptional changes after sorting which would remove any real transcriptional differences between the high and low cells. The lessons learnt from this experiment better prepared me for the experiments in Chapter 2 and also inspired me to look for the effect of cell-state on gene expression noise in Chapter 3.

\section{Scope of thesis work}

This thesis consists of two main chapters. In  Chapter 2 I will try to explain the source of phenotypic variability observed within a cell-type. I will test the hypothesis that cells of the same cell-type express different subsets of genes due to stochastic activation of upstream regulators. Specifically, I will test the hypothesis that stochastic expression of transcription factors and stochastic activation of signaling pathways leads to variability between cells of the same cell-type.

In Chapter 3, I aim to understand what determines the noise levels of different genes in the genome. In collaboration with two other graduate students I developed a new genomic technology called SARGENT (Single-cell Analysis of Reporter Gene Expression Noise and Transcriptome) to study noise across genomic locations. Using SARGENT, we integrate the same reporter gene in multiple genomic locations and measure the cell-to-cell variability in expression of this reporter gene using single-cell RNA sequencing. We then attempt to understand the determinants of this noise, after removing the effect of the mean level of expression, using the chromatin state and sequence features at different genomic locations. We also decompose the variability into extrinsic and intrinsic noise, and understand the determinants of these two noise sources. SARGENT also allows us to identify cell-states and quantify the impact of cell-states on noise.
