\chapter{Effect of genomic and cellular environments on gene expression noise}
\label{chap:cas}

This work was performed in collaboration in with Clarice Hong, Siqi Zhao and Barak Cohen. All authors conceived and designed the project. I performed 10X sample preparation and library preparation, co-developed a pipeline to go from the raw reads to UMI counts for each location in each cell, performed the modeling of mean and noise as a function of covariates. Clarice Hong, Siqi Zhao and I are co-first authors on this manuscript that all authors wrote together. The manuscript is currently under review and has been uploaded to \textit{Biorxiv} (doi: \href{https://www.biorxiv.org/content/10.1101/2022.08.31.506082v1}{10.1101/2022.08.31.506082}) \cite{hongcky_cohenba:EffectGenomic2022}.

\section{Abstract}

Individual cells from isogenic populations often display large cell-to-cell differences in gene expression. This \enquote{noise} in expression derives from several sources, including the genomic and cellular environment in which a gene resides. Large-scale maps of genomic environments have revealed the effects of epigenetic modifications and transcription factor occupancy on mean expression levels, but leveraging such maps to explain expression noise will require new methods to assay how expression noise changes at locations across the genome. To address this gap, we present \underline{S}ingle-cell \underline{A}nalysis of \underline{R}eporter \underline{G}ene \underline{E}xpression \underline{N}oise and \underline{T}ranscriptome (SARGENT), a method that simultaneously measures the noisiness of reporter genes integrated throughout the genome and the global mRNA profiles of individual reporter-gene-containing cells. Using SARGENT, we performed the first comprehensive genome-wide survey of how genomic locations impact gene expression noise. We found that the mean and noise of expression correlate with different histone modifications. We quantified the intrinsic and extrinsic components of reporter gene noise and, using the associated mRNA profiles, assigned the extrinsic component to differences between the CD24+ \enquote{stem-like} sub-state and the more \enquote{differentiated} sub-state. SARGENT also reveals the effects of transgene integrations on endogenous gene expression, which will help guide the search for \enquote{safe-harbor} loci. Taken together, we show that SARGENT is a powerful tool to measure both the mean and noise of gene expression at locations across the genome, and that the data generated by SARGENT reveals important insights into the regulation of gene expression noise genome-wide. 

\section{Introduction}

Gene expression is noisy, even among individual cells from an isogenic population \cite{raja_vanoudenaardena:NatureNurture2008}. Noisy gene expression leads to variable cellular outcomes in differentiation \cite{changhh_huangs:TranscriptomewideNoise2008,kalmart_ariasam:RegulatedFluctuations2009,abranchese_henriqued:StochasticNANOG2014,desairv_weinbergerls:DNARepair2021}, the response to environmental stimuli \cite{spencersl_sorgerpk:NongeneticOrigins2009, topolewskip_komorowskim:PhenotypicVariability2022}, viral latency \cite{weinbergerls_schafferdv:StochasticGene2005}, and chemotherapeutic drug resistance \cite{shaffersm_raja:RareCell2017, emertbl_raja:VariabilityRare2021, yangc_spencersl:MelanomaSubpopulations2021}. Explaining the causes of noisy expression remains an important challenge. 

A gene’s genomic environment, defined here as the composition of nearby cis-regulatory elements and local epigenetic marks, can influence its expression noise. Some features of genomic environments that can affect noise include enhancers, histone modifications, and transcription factor (TF) occupancy \cite{wus_qianw:IndependentRegulation2017, darrd_weinbergerls:TranscriptionalBurst2012, larsondr_singerrh:DirectObservation2013, senecala_darzacqx:TranscriptionFactors2014, waltersmc_martindi:EnhancersIncrease1995, weinbergerl_barkain:ExpressionNoise2012,faureaj_lehnerb:SystematicAnalysis2017}. These observations raise the possibility that genome-wide patterns of expression noise could be explained using the large-scale epigenetic maps that have proved useful in explaining mean expression levels \cite{akhtarw_vansteenselb:ChromatinPosition2013,kundajea_kellism:IntegrativeAnalysis2015,karlicr_vingronm:HistoneModification2010}. Leveraging these resources to explain expression noise will require maps of the genome that show the influence of diverse genomic environments on this noise. Producing these maps will require new experimental approaches, because the existing studies demonstrating the effects of epigenetic marks on expression noise have either been performed on endogenous genes, where the effects of different chromosomal locations are confounded with the effects of the different endogenous promoters or rely on low-throughput imaging methods. Dar \textit{et al.} assayed the noisiness of large numbers of genomic integrations but was unable to assign genomic locations to the measured reporter genes \cite{darrd_weinbergerls:TranscriptionalBurst2012}. Two other studies have assayed integrations in a high-throughput manner but measured protein levels by flow cytometry rather than mRNA levels \cite{deyss_arkinap:OrthogonalControl2015,zhangt_wollmanr:IdentifyingChromatin2020}. Thus, we still lack a high-throughput, systematic way of quantifying the impact of genomic environments on expression noise.

In addition to intrinsic features such as the local genomic environment, extrinsic features, such as the global cellular state of a cell, can also influence gene expression noise \cite{elowitzmb_swainps:StochasticGene2002,ozbudakem_vanoudenaardena:RegulationNoise2002,nevesrp_iborrafj:ConnectingVariability2010,stewart-ornsteinj_el-samadh:CellularNoise2012,  sancheza_goldingi:GeneticDeterminants2013}. For example, variation in the cell cycle, cell size, or signaling pathways can all impact gene expression noise \cite{raja_vanoudenaardena:NatureNurture2008, raserjm_osheaek:ControlStochasticity2004, zopfcj_maheshrin:CellCycleDependence2013}. However, the relative contributions of intrinsic vs extrinsic features on gene expression noise in mammalian cells remains unclear.

Here we report \underline{S}ingle-cell \underline{A}nalysis of \underline{R}eporter \underline{G}ene \underline{E}xpression \underline{N}oise and \underline{T}ranscriptome (SARGENT), a highly parallel method to measure the mean and noise of a common reporter gene that has been integrated at locations across the genome. Analysis of SARGENT data showed that different histone modifications explain the mean and noise produced across the genome. In SARGENT, multiple reporters are integrated in each cell, allowing us to separate the intrinsic and extrinsic contributions to noise. Sequencing the associated single-cell mRNA transcriptomes further enabled us to attribute the extrinsic noise to differences in the cellular substates between isogenic cells. To our knowledge, this is the largest genome-wide survey of the impact of intrinsic and extrinsic noise in gene expression. Taken together, our results show that SARGENT is a powerful tool to study how genomic environments and cellular context control expression noise.

\section{Results}

\subsection{A high-throughput method to measure mean and noise across the genome}

We developed a high-throughput method to test the effects of genomic environments on the mean and noise of gene expression. Our goal was to integrate a common transgene across the genome and then, for individual cells, measure both the transcripts produced from the transgene and the global mRNA profile. This allows us to compute the mean and noise of reporter gene expression at each location, and correlate reporter gene expression with the cellular mRNA state of each cell. Because every unique integration contains the same transgene, the measured differences in the mean and noise of reporter gene expression are directly attributable to the influence of genomic environments or cellular states.

We first generated a reporter gene with a library of 16bp random barcodes (Location barcode, locBC) in its 3’UTR (\fref{fig:cas_figure1}). Due to the diversity of the locBCs, each locBC is only associated with a single location in the genome \cite{akhtarw_vansteenselb:ChromatinPosition2013}. The reporter gene consists of a Cytomegalovirus (CMV) promoter driving the expression of a fluorescent protein and contains a capture sequence from the 10x Genomics Single Cell Gene Expression 3' v3.1 with Feature Barcoding Kit. The 10x gel beads contain both the complementary capture sequence and polyT sequences, allowing us to isolate the transcripts produced from the reporter gene and the cellular transcriptome. 

%figure1
\begin{figure}[t]  
    \centering
    \includegraphics[width=\linewidth]{figures/cas_figure1.png}
    \caption[Overview of the SARGENT workflow.]{%
        \textbf{Overview of the SARGENT workflow.}
         In step 1, a reporter gene driven by the CMV promoter is randomly barcoded with a diverse library of location barcodes (locBC) upstream of the 10x capture sequence (CS). The reporter genes are randomly integrated into K562 cells and sorted for cells with successful integrations (step 2), then sorted again after a week into pools to ensure that each barcode is only represented once per pool (step 3). We then performed scRNA-seq to capture the transcriptome and amplify the expressed barcodes from integrated reporter genes (step 4). The number of expressed barcodes per cell were then tabulated (step 5). To identify the genomic locations of the integrations we also mapped the location of each locBC in bulk (step 5). (ITR = Inverted terminal repeat; prom = promoter)
    }
    \label{fig:cas_figure1}
\end{figure}

To generate chromosomal integrations across the genome we cloned the reporter gene library onto a piggyBac transposon vector. The library was transfected into cells along with piggyBac transposase to allow random integrations of the reporter into the genome. After selecting for integrations in K562 cells, we mapped the locations of each integrated reporter (IR) and assigned each locBC to a specific genomic location. We then captured the reporter gene transcripts from single cells and amplified the barcodes (10x cell barcode, UMI, and locBC) using primers specific to our reporter gene (\hyperref[section:cas_methods]{Methods}). After sequencing and tabulating the mRNA counts for each IR we computed the expression level of the reporter gene at each genomic location in each single cell. For a subset of cells we also sequenced the mRNA profiles to simultaneously reveal the cell state of each individual cell.

\subsection{SARGENT measurements are accurate and reproducible}

We performed SARGENT in K562 cells because of the abundance of public epigenetic data available for this cell line. We first assessed the reproducibility of the SARGENT method. Because replicate infections result in pools of cells with insertions at different genomic locations, we could not assess the reproducibility of independently transfected pools of cells. Instead, we assessed the reproducibility of SARGENT by growing the same pool of insertions twice and performing the SARGENT workflow independently on each sample. We detected 589 identical IR locations in both replicates, which represented 96\% of the total IRs observed in both replicates. After quality control, we obtained data from 7680 single cells across replicates, and a total of 2,940,912 unique molecular identifiers (UMIs) representing expressed barcodes from the IRs in these cells. The replicates were well correlated for measurements of both mean and noise measured at each IR location (\fref{fig:cas_figure2a}-\subcaptionref{fig:cas_figure2b}, mean Pearson’s \textit{r} = 0.76, noise Pearson’s \textit{r} = 0.72) indicating that measurements obtained by SARGENT are reproducible.

%figure2
\begin{figure}[tb]  
    \centering
    \phantomlabel{fig:cas_figure2a}
    \phantomlabel{fig:cas_figure2b}
    \phantomlabel{fig:cas_figure2c}
    \phantomlabel{fig:cas_figure2d}
    \includegraphics[width=\linewidth]{figures/cas_figure2.ai}
    \caption[SARGENT measurements are accurate and reproducible.]{%
        \textbf{SARGENT measurements are accurate and reproducible.}
        \subref{fig:cas_figure2a}
        Correlation of mean levels between technical replicates.
        \subref{fig:cas_figure2b}
        Correlation of variance measurements between replicates.
        \subref{fig:cas_figure2c}
        Mean and variance are correlated within each experiment.
        \subref{fig:cas_figure2d}
        Mean-independent noise corrects for mean effects on variance. Correlations shown are Pearson’s correlation coefficients (Pearson’s \textit{r}). 
    }
    \label{fig:cas_figure2}
\end{figure}

To validate the single-cell measurements made by SARGENT, we also performed single-molecule Fluorescence In Situ Hybridization (smFISH) on two known locations. The measurements of mean and noise made by smFISH agree with the SARGENT measurements for those locations (\fref{fig:cas_figureS1}) demonstrating that our method is accurate and reproducible for measuring the mean and noise of expression. 

\subsection{Measurements of mean-independent noise across different chromosomal environments}

In total, we performed four experiments and generated mean and noise measurements for 939 integrations. The integrations were spread across the genome and found in regions with different chromHMM annotations \cite{ernstj_kellism:DiscoveryCharacterization2010} (\fref{fig:cas_figureS2a}-\subcaptionref{fig:cas_figureS2b}), allowing us to study the effects of diverse chromosomal environments on expression noise.

The mean and noise of expression are often highly correlated \cite{vallaniaflm_mitrard:OriginConsequences2014,bar-evena_barkain:NoiseProtein2006}. Similarly, we found a strong correlation between the mean and noise in SARGENT data, indicating that a large proportion of an IR’s noise is explained by its mean level of expression (\fref{fig:cas_figure2c}). To identify chromosomal features that control expression noise independent of mean levels we regressed out the effect of mean levels on noise, leaving us with a metric we refer to as mean-independent noise (MIN) \cite{vallaniaflm_mitrard:OriginConsequences2014}. By design, MIN levels of IRs are uncorrelated with their mean expression levels (\fref{fig:cas_figure2d}) whereas other measures of noise, such as the coefficient of variance or the Fano factor, retain residual correlation with mean levels in our data (\fref{fig:cas_figureS2c}-\subcaptionref{fig:cas_figureS2d}). Thus, we used MIN as a measure of expression noise for all following analyses.

\subsection{Expression mean and noise are associated with different chromosomal features}

We sought to identify chromatin features that would explain differences in MIN levels between genomic locations. Studies of genome-wide chromatin features in many cell lines and tissues have shown that the mean expression of a gene is correlated with its surrounding chromatin marks \cite{dunham2012n, akhtarw_vansteenselb:ChromatinPosition2013}. Thus, we asked whether chromatin features might also explain patterns of MIN across the genome. We split the IRs into bins of high or low mean levels, or high or low MIN levels, and identified chromatin features that were enriched in specific bins. As expected, IRs with high mean expression had higher levels of active chromatin marks such as H3K27ac, H3K4 methylation, H3K79me2 and H3K9ac (\fref{fig:cas_figure3a}). Conversely, IRs with high MIN did not exhibit significant differences between H3K27ac or H3K4me1 levels, and low MIN locations showed slightly elevated levels of H3K4me2/3, H3K79me2 and H3K9ac (\fref{fig:cas_figure3b}). These results suggest that different chromatin modifications influence the mean and noisiness of expression, and that more active genomic locations might also reduce MIN. This observation is consistent with previous studies showing that repressed chromatin is associated with high MIN \cite{faureaj_lehnerb:SystematicAnalysis2017,deyss_arkinap:OrthogonalControl2015}. 

%figure3
\begin{figure}[tbp]  
    \centering
    \phantomlabel{fig:cas_figure3a}
    \phantomlabel{fig:cas_figure3b}
    \phantomlabel{fig:cas_figure3c}
    \phantomlabel{fig:cas_figure3d}
    \includegraphics[width=\linewidth]{figures/cas_figure3.ai}
    \caption[Chromosomal features associated with expression mean and noise.]{%
        \textbf{Expression mean and noise are associated with different chromosomal features.}
        \subref{fig:cas_figure3a}
        Active histone modifications associated with high or low mean IRs. Start indicates the location of the IR, and each location was extended 5kb on either side.
        \subref{fig:cas_figure3b}
        Active histone modifications associated with high or low MIN IRs are different from those associated with mean.
        \subref{fig:cas_figure3c}
        Motifs enriched in high or low MIN IRs respectively, and potential TFs that match these discovered motifs.
        \subref{fig:cas_figure3d}
        Logistic regression weights of various intrinsic features associated with high or low MIN IRs. Red bars: \textit{p}-value $<$ 0.05; Pink bars: 0.05 $<$ \textit{p}-value $<$ 0.1 from the logistic regression model.  
    }
    \label{fig:cas_figure3}
\end{figure}

The binding of TFs also impacts noise in gene expression. To identify TFs that might affect noise, we identified TFs whose occupancy is enriched near either high or low MIN IRs. Sequences at low MIN IRs are enriched for transcriptional activators such as SP1 and E2F4, while sequences at high MIN IRs are enriched for other TFs including TFs containing basic helix-loop-helix (bHLH) domains (\fref{fig:cas_figure3c}), suggesting that the cofactors recruited by different TFs have separable effects on expression mean and noise. 

To assess the power of genomic features to predict the MIN of IR locations we trained a logistic regression model using chromatin modifications and DNA sequence features to classify high and low MIN locations and achieved an accuracy of 76\%. When applied to data from a different pool, the trained model achieved 67\% accuracy. The features with significant weights are the H3K4me3 mark, TF motifs (RARG, FOXO4, HIF1A, TFAP4, CREM, ATF1, NFIC, and NFIA) and whether the IR location was inside a gene (\fref{fig:cas_figure3d}). Being inside a gene reduced the probability of being a high noise lR location, which could be due to local regulatory elements that dampen the noisiness of a gene’s expression. Similar to our results above, lower H3K4me3 increased the probability of being a high noise IR location. H3K4me3 is associated with active chromatin and supports the hypothesis that higher activity reduces IR MIN. Our observation is consistent with a previous study showing that H3K4me3 correlates with reduced noise at endogenous genes \cite{faureaj_lehnerb:SystematicAnalysis2017}. With respect to the effects of TFs on noise, the presence of some TF motifs increase the probability of being a high noise IR location (NFIC, CREM, TFAP4, CLOCK), whereas other TFs reduce the probability of being a high noise location (RARG, NFIA, ATF1, FOXO4, HIF1A). 

We used a similar logistic regression framework to identify features that separate IR locations with high or low mean levels of expression and achieved an accuracy of 82\%. When applied to holdout data, the trained model achieved 62\% accuracy. The chromatin features that increase the probability of being a high mean IR location are lower levels of H3K27me3, lower levels of H3K4me2 and a higher number of ATAC-seq peaks, which agrees with the known effects of these features in bulk mean expression. The motifs that increased the probability of being a high mean IR location are higher numbers of motifs of the ZNF76, BACH1 and E2F3 TFs and fewer instances of the E2F7, SMAD3 and SOX5 motifs (\fref{fig:cas_figureS3}). Comparisons of the models explaining either mean or noise again show that different genomic features are correlated with gene expression mean and noise.

\subsection{Intrinsic and extrinsic factors have similar effects on gene expression noise}

Expression noise caused by fluctuations in global factors affects all genes and is referred to as extrinsic noise, whereas intrinsic sources of noise are specific to individual genes \cite{deyss_arkinap:OrthogonalControl2015, stewart-ornsteinj_el-samadh:CellularNoise2012, sancheza_goldingi:GeneticDeterminants2013, raserjm_osheaek:ControlStochasticity2004, zopfcj_maheshrin:CellCycleDependence2013, vallaniaflm_mitrard:OriginConsequences2014}. The correlation between identical reporter genes in the same cell measures the balance between extrinsic and intrinsic noise, with extrinsic factors increasing the correlation \cite{elowitzmb_swainps:StochasticGene2002}. In SARGENT, the correlation between IRs in the same cells is a measure of extrinsic factors that affect noise across IR locations. 

For our analysis of extrinsic noise we first identified IRs in the same clonal cells using the co-occurrence of locBCs between single cells (\fref{fig:cas_figure4a}). We identified 192 clones, with a mean of three integrations per clone (\fref{fig:cas_figureS4a}). Of these 192 clones, 45 contain more than one integration (\fref{fig:cas_figure4b}), making them suitable for an analysis of extrinsic noise. To validate the identified clones, we individually mapped IR barcodes in sixteen clones and found that 94\% of the individually mapped IR locations could be uniquely assigned to an identified clone (\fref{fig:cas_figure4b}). 

%figure4
\begin{figure}[tbp]  
    \centering
    \phantomlabel{fig:cas_figure4a}
    \phantomlabel{fig:cas_figure4b}
    \phantomlabel{fig:cas_figure4c}
    \phantomlabel{fig:cas_figure4d}
    \includegraphics[width=\linewidth]{figures/cas_figure4.ai}
    \caption[SARGENT quantifies the extrinsic portion of expression noise.]{%
        \textbf{SARGENT quantifies the extrinsic portion of expression noise.}
        \subref{fig:cas_figure4a}
        Schematic for identifying different initial TRIP clones.
        \subref{fig:cas_figure4b}
        A network representation of the different clones identified, red nodes indicate IR locations that were independently validated by sequencing individual clones.
        \subref{fig:cas_figure4c}
        Pairwise expression for any two IR locations observed in the same cell. The trend along the diagonal suggests the existence of extrinsic noise, and the anti-correlation indicates the amount of intrinsic noise.
        \subref{fig:cas_figure4d}
        Quantification of intrinsic and extrinsic proportion of noise. Error bars from two technical replicates. 
    }
    \label{fig:cas_figure4}
\end{figure}

We next asked if extrinsic factors also contribute to the observed gene expression noise. For each cell in a clone, we calculated the standard deviation relative to the mean of all IRs in that cell, which we define as the fluctuation index. Lower fluctuation indices indicate that the IRs in a clone fluctuate in sync (high extrinsic noise), while higher fluctuation indexes indicate that each IR varies independently (high intrinsic noise). To simulate intrinsic noise, we first shuffled the cell labels of all the IRs within a clone and computed a distribution of fluctuation indexes for the shuffled population. If all the measured noise was intrinsic, then the measured distribution would perfectly overlap the shuffled distribution. If all the measured noise was extrinsic, then all the cells would have fluctuation indexes of 0 (\fref{fig:cas_figureS4b}). We found that all clones show a distribution of fluctuation indexes that is lower than that of the shuffled distribution and above zero (\fref{fig:cas_figureS4c}). This suggests that some portion of the expression noise can be explained by extrinsic factors that impact all IRs within a cell in different genomic environments.

To quantify the contribution of intrinsic and extrinsic noise in each clone we employed an established statistical framework \cite{fuaq_pachterl:EstimatingIntrinsic2016}. Using the pairwise IR single cell expressions for all clones that contain more than one IR as input, we found that intrinsic noise comprises approximately 54\% of the total noise (\fref{fig:cas_figure4c}-\subcaptionref{fig:cas_figure4d}). This analysis suggests that both the intrinsic chromatin and extrinsic cellular context explains about half of the total noise in each clone. These results show that SARGENT can quantify both intrinsic and extrinsic contributions to expression noise.

\subsection{Cell substates are a source of expression noise}

What cellular mechanisms control expression noise? We hypothesized that differences between cellular substates within isogenic populations are an important source of noise. Isogenic K562 cells transition between \enquote{stem-like} and \enquote{more differentiated} substates \cite{litzenburgerum_changhy:SinglecellEpigenomic2017, moudgila_mitrard:SelfReportingTransposons2020}. The stem-like substate is marked by high CD24 expression and proliferates at a higher rate, which we hypothesized would contribute to extrinsic noise. This hypothesis predicts that the same IRs will have higher MIN in stem-like cells compared to more differentiated cells. To test this prediction we sequenced the single-cell transcriptomes associated with 356 of the 939 genomic locations in parallel with the IRs. Using the transcriptomes we identified clusters of cells with high CD24 expression and confirmed that these clusters had the signatures of high-proliferating cells (\fref{fig:cas_figureS5a}-\subcaptionref{fig:cas_figureS5b}). We then calculated the expression mean and MIN for each IR location separately in the two substates. IR locations in the stem-like substate have higher mean and lower MIN (\fref{fig:cas_figure5a}-\subcaptionref{fig:cas_figure5b}) suggesting that the global differences between the two substates are a source of MIN. 

%figure5
\begin{figure}[tbp]  
    \centering
    \phantomlabel{fig:cas_figure5a}
    \phantomlabel{fig:cas_figure5b}
    \phantomlabel{fig:cas_figure5c}
    \phantomlabel{fig:cas_figure5d}
    \phantomlabel{fig:cas_figure5e}
    \phantomlabel{fig:cas_figure5f}
    \phantomlabel{fig:cas_figure5g}
    \phantomlabel{fig:cas_figure5h}
    \includegraphics[width=\linewidth]{figures/cas_figure5.ai}
    \caption[Cellular information improves classification of MIN IR locations.]{%
        \textbf{Cellular information improves classification of low vs high MIN IR locations. (A,B)}
        Violin plots of expression mean \subref{fig:cas_figure5a} and MIN \subref{fig:cas_figure5b} at two substates (student t-test, $\ast\ast\ast\ast$: \textit{p} <  0.0001), each dot is an IR location.
        \textbf{(C,D)}
        Scatterplots of proportion of cells in the \enquote{stem-like} substate against mean \subref{fig:cas_figure5c} and MIN \subref{fig:cas_figure5d}. Each dot is the average mean expression or MIN from a clone. Line: linear fit with 95\% confidence interval. 
        \legendcontdnote
    }
    \label{fig:cas_figure5}
\end{figure}

\begin{figure}[t]
    \centering
    \legend{%
        \legendcontdref{fig:cas_figure5}
        \subref{fig:cas_figure5e}
        Barplot of the fraction of cells in different cell cycle phases for cells in the \enquote{stem-like} substate and the \enquote{differentiated} substate (Binomial test: S phase \textit{p} $<$ 2.2e-16, G1 phase \textit{p} $<$ 5.9e-5, G2M phase \textit{p} $<$ 2.2e-16). The error bars are derived from the two replicates.
        \subref{fig:cas_figure5f}
        Weights of logistic regression model using extrinsic (cellular) features alone.
        \subref{fig:cas_figure5g}
        Addition of extrinsic features helps to improve the accuracy of the model.
        \subref{fig:cas_figure5h}
        Weights of logistic regression model using both intrinsic and extrinsic features. The most significant features are still the proportion of cells in the G2 phase and CD24+ phase. Red bars: \textit{p}-value $<$ 0.05; Pink bars: 0.05 $<$ \textit{p}-value $<$ 0.1 from the logistic regression model. 
    }
\end{figure}

Given the differences in mean and MIN between the substates, the MIN of the IR locations in a given clone should be partly explained by the proportion of its cells in each substate. Consistent with this prediction, we found that clones with a higher proportion of cells in the stem-like substate have slightly higher average mean expression (Spearman’s \textit{ρ} = 0.22, \textit{p}-value = 0.008), and lower average MIN (Spearman’s \textit{ρ} = -0.27, \textit{p}-value = 0.0015) across all IRs in the clone (\fref{fig:cas_figure5c}-\subcaptionref{fig:cas_figure5d}). We hypothesized that this was due to the slightly higher proliferation rates of cells in the stem-like phase. As expected, there are more cells in the S phase in the stem-like substate compared to the more differentiated state (\fref{fig:cas_figure5e}). We then examined the differences of mean and MIN in different cell cycle phases and found that expression mean is higher and MIN is lower in the S phase compared to other phases (\fref{fig:cas_figureS5c}-\subcaptionref{fig:cas_figureS5d}). These results suggest that differences in proliferation rates is an important source of extrinsic noise, and that SARGENT is a powerful tool to dissect the extrinsic sources of expression noise. 

\subsection{Cellular information improves classification of low vs high MIN IR locations}

Since extrinsic factors play an important role in determining expression noise, we trained a logistic regression model to predict MIN using three extrinsic features (proportion of cells in S, proportion of cells in G2, and proportion of CD24+ cells). Using only the global features, the model achieved 75\% accuracy (albeit without a holdout set to test on due to the small numbers of locations with associated extrinsic features). This result implies that these cellular features explain a significant portion of the variance in MIN between high and low IR locations. The proportion of cells in G2 and the proportion of cells in the CD24+ state were significant predictors in this model. Being in G2 increases the probability of a high MIN IR location whereas having a higher proportion of CD24 cells reduced the probability of being a high MIN IR location (\fref{fig:cas_figure5f}). When we combined the significant intrinsic features from the previous model with these extrinsic features, the model accuracy increased to 84\% (\fref{fig:cas_figure5g}). In the combined model, the extrinsic features have higher weights than the intrinsic genomic environment features (\fref{fig:cas_figure5h}), suggesting that the cell-state information may play a larger role in regulating MIN compared to genomic environments. 

We observed a similar role for extrinsic features in classifying IR locations with high mean levels from IR locations with low mean levels. The model accuracy for just the extrinsic feature model is 80\% and increases to 89\% for the combined model with both intrinsic and extrinsic features (\fref{fig:cas_figureS5e}). In the combined model, the proportion of cells in the CD24 cell-state is the most highly weighted feature (\fref{fig:cas_figureS5f}). In contrast to the MIN model, the proportion of cells in the CD24 state increases the probability of being a high-mean IR location (\fref{fig:cas_figure5h}, \fref{fig:cas_figureS5f}), which is consistent with our observations in \fref{fig:cas_figure5b} and \subcaptionref{fig:cas_figure5d}. Thus, while cellular information plays an important role in gene expression regulation, these features have orthogonal impacts on expression mean and single-cell variability. 

\subsection{Effects of transgenes integration on endogenous genes}

Finally, SARGENT can be used for purposes beyond studying gene expression noise. One such application is screening for \enquote{safe harbor} loci in the genome. To achieve safe and effective gene therapy, we need to identify genomic locations that have stable expression of the transgene of interest (high mean expression and low noise) and have minimal effects on endogenous gene expression. Historically, transgenes are often integrated into several known \enquote{safe harbor} loci \cite{aznauryane_reddyst:DiscoveryValidation2022}. Those loci are mainly located in the introns of stably expressed genes to prevent silencing. Because SARGENT can be used to measure gene expression mean, noise and endogenous gene expression simultaneously, we can leverage SARGENT to screen for potential safe harbors in a high-throughput manner. 

We examined how our reporter gene integrations altered the expression of the gene into which it integrated. We focused on the 65 IR locations that are integrated into gene bodies. These integrations were distributed across different clones (\fref{fig:cas_figureS6a}) and should not be confounded by clonal effects. We calculated pseudo-bulk expression for each gene from clones that contain the integration and compared that to the expression from other clones that do not have the IR integration (\fref{fig:cas_figure6a}). We found that in most cases (61/65), transgene integration does not alter the endogenous gene expression (\fref{fig:cas_figure6b}). We also randomly shuffled the gene labels to compute the background differential expression, and found that there were no significantly differentially expressed genes once the labels were shuffled (\fref{fig:cas_figureS6b}). Among the locations with significantly differentially expressed genes, 3 out of 4 IR integrations increases gene expression (\fref{fig:cas_figure6c}), consistent with previous studies showing that the integration of a transgene often increases endogenous gene expression \cite{papapetrouep_schambacha:GeneInsertion2016}. Taken together, our results suggest that most endogenous genes are not impacted by the integration of exogenous genes. This result illustrates that SARGENT could be a powerful tool to screen for \enquote{safe harbor} loci for transgene integration.

%figure6
\begin{figure}[tbp]  
    \centering
    \phantomlabel{fig:cas_figure6a}
    \phantomlabel{fig:cas_figure6b}
    \phantomlabel{fig:cas_figure6c}
    \includegraphics[width=\linewidth]{figures/cas_figure6.ai}
    \caption[SARGENT measures the insertion effect of a transgene.]{%
        \textbf{SARGENT measures the insertion effect of a transgene.}
        \subref{fig:cas_figure6a}
        Schematic for expression change detection in the transcriptome data.
        \subref{fig:cas_figure6b}
        Volcano plot of log\textsubscript{2} fold change and -log\textsubscript{10}(\textit{p}-value) from a Fisher’s Exact Test. Red dotted lines: cut off for fold change (0.5), cut off for \textit{p}-value (0.05).
        \subref{fig:cas_figure6c}
        Barplots of difference of expression between genes without IRs (control) and genes with IRs (insert). The clone where the IR is integrated is indicated. Error bars are derived from two technical replicates.
    }
    \label{fig:cas_figure6}
\end{figure}

\section{Discussion}

Since the early single-cell studies showing the variability of gene expression in isogenic populations \cite{elowitzmb_swainps:StochasticGene2002}, many individual chromatin and sequence features have been suggested to modulate expression noise \cite{raja_vanoudenaardena:NatureNurture2008, raja_tyagis:StochasticMRNA2006,bonnyar_el-samadh:OrthogonalControl2021, desairv_weinbergerls:DNARepair2021}. However, there has yet to be a systematic study of the impact of different genomic features on large numbers of identical genes. 
   
We developed SARGENT, a high throughput method to measure the expression mean and noise at different genomic locations in parallel. One key advantage of SARGENT is that the reporter gene used in all locations is identical, which allows us to isolate the effects of the genomic environments without being confounded by the effects of different promoters. We identified different chromatin marks that are associated with high or low MIN, and used a logistic regression model to identify features of the genomic environments that might control MIN. Our observations indicate that the features that control expression noise are independent of the features controlling expression mean. Several recent studies have developed tools for the orthogonal control of mean and gene expression noise \cite{bonnyar_el-samadh:OrthogonalControl2021,benzingerd_khammashm:PulsatileInputs2018, michaelsys_fulgata:PreciseTuning2019}. To this end, our results suggest potential mechanisms that can be targeted for independent modulation of expression mean and single-cell variability. 

We also quantified the extrinsic portion of expression noise and identified that the oscillation between a \enquote{stem-like} substate and a \enquote{differentiated} substate in K562 cells is an important source of extrinsic noise. Our data suggests that extrinsic noise might be more important in regulating MIN than genomic environments. This indicates that the regulation of noise of individual genes might be at the level of the promoter, rather than through its chromatin or genomic environment.
 
We envision that SARGENT will be a useful tool for other synthetic biology applications. While advances in genome engineering technologies now allow researchers to integrate transgenes at most desired genomic locations, the selection of appropriate sites for transgene overexpression remains non-trivial, with no location in human cells validated as a safe harbor locus \cite{papapetrouep_schambacha:GeneInsertion2016,pavanig_amendolam:TargetedGene2021}. This is mainly due to the lack of methods to systematically screen for loci that have high expression, low variability and do not impact cellular function. Here we showed that SARGENT can be used to read out a transgene’s impact on global expression as well as the endogenous gene that it is integrated into. Surprisingly, most endogenous genes are not impacted by the insertion of reporter genes. With SARGENT, we can quickly screen genomic locations to find the best locations for human transgene integration which will prove useful gene therapy applications.

More broadly, we envision that SARGENT will be a useful technology for many different applications including mechanistic studies of gene expression noise and synthetic biology applications. The 10x Genomics platform used in this study is limited by throughput, but improvements to scRNA-seq technologies will increase the scope of SARGENT. For example, coupling sci-RNA-seq \cite{caoj_shendurej:ComprehensiveSinglecell2017} or SPLiT-seq \cite{rosenbergab_seeligg:SinglecellProfiling2018} to SARGENT would allow for many more locations to be assayed in parallel. A larger goal will be to construct a detailed map of the MIN landscape across the genome, much like the maps of mean expression levels generated by ENCODE. 

\section{Methods}
\label{section:cas_methods}

\subsection*{SARGENT library cloning}

All primers and oligonucleotides used in this study are listed in \tref{tab:cas_tableS1}. To clone the reporter gene for SARGENT, we first cloned a CMV-BFP reporter gene containing the 10x capture sequence 1 (CS1) into a piggyBac vector containing two parts of a split-GFP reporter gene \cite{qiz_mitrard:OptimizedBroadly2017}. When the reporter gene construct is integrated into the genome, the split-GFP combines to produce functional GFP, allowing us to sort for cells that have successful reporter gene integrations. We next added a library of random barcodes to the plasmid by digesting the plasmid with XbaI followed by NEBuilder HiFi DNA Assembly (NEB \#E2621) with a single-stranded oligo containing 16 random N’s (Location barcodes; locBC) and homology arms to the plasmid (CAS P57). 

\subsection*{Generation of cell lines for SARGENT}

K562 cells were maintained in Iscove's Modified Dulbecco′s Medium (IMDM) + 10\% FBS + 1\% non-essential amino acids + 1\% penicillin/streptomycin. We selected two K562 cell lines previously used in our lab that each contain a \enquote{landing pad} at a unique location with a pair of asymmetric Lox sites for recombination (loc1 - chr8:144,796,786, loc2 - chr11: 16,237,204; hg38 coordinates). Using these \enquote{landing pad} cell lines allows us to perform smFISH on the landing pad to directly compare SARGENT and smFISH results. For each cell line, we replaced the original landing pad cassette with the same reporter gene in the SARGENT library so that we can capture the reporters from the landing pad and reporters from other genomic locations in SARGENT using the same primers. Pool 1 was derived from the loc2 cell line, while Pools 2, 3 and 4 were derived from the loc1 cell line.

The SARGENT library and piggyBac transposase were co-transfected into K562 (LP cell lines) cells at a 3:1 ratio using the Neon Transfection System (Life Technologies). For each experiment, we transfected 2.4 million cells with 9μg of SARGENT library and 3μg of transposase. The cells were sorted after 24 hours for GFP-positive cells to enrich for cells that have integrated SARGENT reporters. We reasoned that \textasciitilde100 single cells for each Integrated Reporter (IR) location would be required to obtain a good estimate of mean and variance. Each SARGENT experiment contains many single cell clone expansions: all the cells from the same clone share the same genomic integrations. Since we targeted approximately 20,000 cells per 10x run, the upper limit of the numbers of clones we can test in one experiment is 200. Because 10x also has a high dropout rate, we targeted 100 clones per experiment in order to ensure that we obtained high quality data. Each clone has an average of 5 integrations, which theoretically allows us to assay 500 IR locations in one experiment. Since the clones did not all grow at the same rate, practically we obtained fewer than 500 IRs per experiment.

For Pools 1 and 2, cells were sorted into pools of 100 cells each and allowed to grow until there were sufficient cells for RNA/DNA extraction and SARGENT experiments. Pool 3 contained the same cells as Pool 2, except that single cells were allowed to grow individually and pooled by hand just before the SARGENT experiments. This allowed for a more even representation of each individual clone (which contains unique integrations) in the final pool. For Pool 4, transfected cells were first sorted into 96-well plates with 2 cells/well and allowed to grow individually and 100 wells were manually pooled for SARGENT experiments. We used cells from Pool 4 to compute technical reproducibility.

\subsection*{SARGENT integration mapping}

We harvested DNA from SARGENT pools using the TRIzol reagent (Life Technologies). To map the locations of SARGENT integrations, we digested gDNA for each pool with a combination of AvrII, NheI, SpeI and XbaI for 16 hours. The digestions were purified and self-ligated at 16\dg C for another 16 hours. After purifying the ligations, we performed inverse PCR to amplify the barcodes with the associated genomic DNA region (CAS P59 and P64). For each pool, we performed 2 technical replicates with 8 PCRs per replicate and pooled the PCRs of each replicate for purification. We then used 8ng of each replicate for further amplification with 2 rounds of PCR to add Illumina sequencing adapters (CAS P55 and P65). The sequencing library was sequenced on the Illumina NextSeq platform. 

The barcodes of each read were matched with the sequence of its integration site. The integration site sequences were then aligned to hg38 using BWA with default parameters \cite{lih_durbinr:FastAccurate2009}. Only barcodes that mapped to a unique location were kept for downstream analyses. 

\subsection*{ClampFISH}
 
Single-molecule FISH was performed on the two \enquote{landing pad} locations that were in the original cell lines used for SARGENT (see Generation of cell lines for SARGENT above). ClampFISH probes for the reporter genes were designed using the Raj Lab Probe Design Tool. Each probe was broken into three arms to be synthesized by IDT. The 5’ of the left arm is labeled by a hexynyl group, and the 3’ of the right arm is labeled by NHS-azide. The right arm fragment was purified by HPLC. All three components were resuspended in nuclease-free H\textsubscript{2}O to a concentration of 400μM. The three arms were ligated by T7 ligase (NEB \# M0318), at 25\dg C overnight. then purified using the Monarch PCR \& DNA cleanup Kit (NEB \#T1030) and eluted with 40μl of nuclease-free water. After the ligation, each probe is stored at -20\dg C. ClampFISH was performed according to the suspension cell line protocol of clampFISH \cite{rouhanifardsh_raja:ClampFISHDetects2019}. 0.7 million cells were collected and fixed in 2mL of fixing buffer containing 4\% formaldehyde for 10 min, then permeabilized in 70\% EtOH at 4\dg C for 24 hours. The primary ClampFISH probes were then hybridized for 4 hours at 37\dg C in the hybridization buffer (10\% Dextran Sulfate, 10\% Formamide, 2X SSC, 0.25\% Triton X). After hybridization, cells were spun down gently at 1000 rcf for 2 min. Cells were washed twice with the washing buffer (20\% formamide, 2X SSC, 0.25\% Triton X) for 30 min at 37\dg C. The secondary probes were then hybridized to cells at 37\dg C for 2 hours and the cells were then washed twice with washing buffer for 30 min at 37\dg C. The primary and secondary probes are \enquote{clamped} in place through a click reaction (CuSO4 75μM, BTTAA 150μM, Sodium Ascorbate 2.5 mM in 2X SSC) for 20 min at 37\dg C. The cells were then washed twice in the washing buffer at 37\dg C for 30 min each wash. Then, the cells were hybridized with the hybridization buffer with tertiary probes for 2 hours at 37\dg C. We complete 6 cycles of hybridization for all our experiments. After the final washes, cells were incubated at 37\dg C with 100mM DAPI for 20 min, washed twice with PBS, resuspended in the anti-fade buffer, and spun onto a \#1.5 coverslip using a cytospin cytocentrifuge (Thermo Scientific), mounted onto a glass slide, sealed with a sealant, and stored at 4\dg C. 

\subsection*{SARGENT library using the 10x Genomics platform}

\subsection*{Cell preparation}

We used the Chromium Single Cell 3’ Kit (v3.1) from 10x Genomics for SARGENT. We followed the manufacturer’s instructions for preparing single-cell suspensions. We used a cell counter to measure the number of cells and viability, and used cell preparations with greater than 95\% cell viability. 

\subsection*{Cell barcoding and reverse transcription}

We followed the manufacturer’s instructions with the following modifications in Pools 1-3: no 10x template switching oligo (PN3000228) was added to the Master Mix (Step 1.1). To correct for the missing volume, 2.4μl of H2O was added to the master mix per reaction. For pool 4, the template switching oligo was included as written. For the cDNA amplification (Step 2.2), no 10x provided reagents were used. Instead, a custom primer (CAS P20) was used with 14 cycles of amplification with the provided 10x protocol (Step 2.2 d). For the pool where we also sequenced transcriptomes (Pool 4), we followed the 10x protocol as written for cDNA amplification. 

\subsection*{Barcode PCR and library preparation}

We performed nested PCRs to amplify barcodes from 10x cDNA. For pools 1-2, PCR library construction was split into two pools for amplification of transcripts captured by capture sequence 1 and poly(A) respectively. Both PCR reactions were done with 2μl purified cDNA, 2.5μl 10μM reporter-specific forward primer (CAS P45), 2.5μl 10 uM poly(A) (CAS P20) or capture sequence adapter-specific primers (CAS P32), and 25μl Q5 High Fidelity 2X Master Mix (NEB \#M0492) in 50μl total volume with 10 cycles amplification. The PCRs were then purified with Monarch PCR \& DNA Cleanup Kit (NEB \#T1030) and Illumina adapters were added in another 2 rounds of PCR, with a PCR purification step with the Monarch kit between PCRs. For poly(A) amplicons, we used CAS P42 and CAS PP2, followed by CAS P48 and CAS PP4. For capture sequence amplicons, we used CAS P41 and CAS CS2, followed by CAS P48 and CAS CS4. The reactions were then pooled and purified with SPRIselect Beads (Beckman Coulter) at 0.65x volume. For pool 4, we performed the PCRs for the poly(A) fraction using 2μl purified cDNA as described above, but not the capture sequence transcripts.

\subsection*{SARGENT data processing}
\subsection*{Read parsing}

We first identified the reads that match the constant sequence in our reporter gene. We used two versions of constant sequence to match against, depending on if the read was captured using the poly(A) sequence on the mRNA or the capture sequence specific to the 10x beads. We used a fuzzy match algorithm to capture reads that have a mismatch at these positions due to sequencing error. From each read, we parsed out the cell barcode, 10x UMI and locBC. We then collapsed reads with identical cell barcodes, UMI and locBCs into one \enquote{trio} and kept track of the number of reads supporting each trio. For downstream analysis, we filtered out trios with low numbers of supporting reads since these are likely to be enriched for PCR artifacts. We next processed the trios to error correct the cell barcodes and locBCs before estimating the mean and variance. 

\subsection*{Barcode error correction}

To correct for PCR artifact and sequencing errors, a custom script was used to error-correct for 10x cell barcodes. Briefly, we first acquired the empirical distribution of the Hamming distances among observed 10x cell barcodes. We found that more than 99\% of 10x cell barcodes are more than 6 hamming distances away from each other, making error correction a feasible approach to denoise the data. We first identify cell barcodes that match perfectly to the 10x cell barcode whitelist, then we order them based on their abundance. The cell barcodes that are not in the whitelist are then compared to the ordered whitelisted cell barcodes, if the Hamming distance between the non-whitelisted cell barcodes is within 2 Hamming distances of a whitelisted cell barcodes, we correct the unwhitelisted cell barcode. With cell barcode correction, we recovered \textasciitilde12\% of reads that would have been discarded. 

Due to the random synthesis of the locBC, a slightly different approach was taken for error correction for the locBCs. Briefly, all the locBCs are ranked based on abundance. Starting from the most abundant barcode, we look for locBCs that are within 4 Hamming distance to that barcode and correct them. We then remove that barcode and any corrected barcodes, and repeat this process until we have iterated through all locBCs.

\subsection*{Calculating mean and variance of each IR}

We filtered out cells that had less than 5 IR integrations (locBCs) and less than 10 UMIs. We also filtered out locBCs that were seen in less than 5 cells and UMIs that had less than 2 supporting reads. We then computed the number of UMIs per locBC in each cell to calculate the expression level of each locBC. For each locBC, mean expression was calculated as the average UMI count across all cells that expressed that locBC. Expression variance was calculated as the variance in UMI counts across all cells that expressed that locBC. 

\subsection*{Mean-independent noise (MIN) metric}

In order to remove the effect of the mean on the variance we first fit a linear model: $\log_2(\text{variance of IR location}) \sim \log_2(\text{mean of IR location})$ for each experiment al pool and used the residuals of the model as the mean-independent noise metric. For each IR location, the MIN is the residual variance after removing the effect of the mean.
Analyses of genomic environment effects on mean-independent noise

\subsection*{Chromatin environment association with mean/MIN}

We downloaded the Core 15-state chromHMM annotations for K562 cells from the Roadmap Epigenomics Project \cite{kundajea_kellism:IntegrativeAnalysis2015}. We then collapsed similar annotations and overlapped the IR locations with the corresponding annotation using the GenomicRanges R package \cite{lawrencem_careyvj:SoftwareComputing2013}. 

We split the IRs into locations with high vs low mean or high vs low MIN respectively. We then downloaded histone ChIP-seq datasets from ENCODE \cite{mooreje_wengz:ExpandedEncyclopaedias2020} and plotted the signals 10kb surrounding each class of IRs using the ComplexHeatmap package in R \cite{guz_schlesnerm:ComplexHeatmaps2016}. 

To look for enriched TF motifs we first downloaded all human motifs from the HOCOMOCO v11 database. We then filtered the motifs for TFs that are expressed (FPKM $\geq$ 1) in the K562 cell line using whole-cell long poly(A) RNA-seq data generated by ENCODE (downloaded from the EMBL-EBI Expression Atlas). We then used the STREME package \cite{baileytl_baileytl:STREMEAccurate2021} (MEME suite 5.4.1) with sequences of 1kb surrounding each IR to identify enriched de novo motifs in high or low MIN regions, using the other class as the control set of sequences. We then took the top 2 motifs for each bin and matched it against a list of TFs expressed in K562s using TOMTOM \cite{guptas_noblews:QuantifyingSimilarity2007} (MEME suite 5.4.1). We reported the top 6 TOMTOM matches. 

\subsection*{K562 Hi-C}

We performed Hi-C on wild-type K562 cells with the Arima Hi-C kit (A510008) according to the manufacturer's protocols (3 replicates, 870 million reads total). The reads were then processed with the Juicer pipeline \cite{durandnc_aidenel:JuicerProvides2016} to generate HiC contact files for each replicate. We then used the peakHiC tool \cite{bianchiv_laatw:DetailedRegulatory2019} to call loops from each IR with the following parameters: window size = 80, alphaFDR = 0.5, minimum distance = 10kb, qWr = 1. Using these parameters each IR was looped to a median of 3 regions (range 0-7). 

\subsection*{Logistic regression model for intrinsic and extrinsic features associated with MIN}

We used histone ChIP-seq and ATAC-seq datasets from ENCODE \cite{dunham2012n} and overlapped their signals with each IR using used bedtools v2.27.1 \cite{quinlanar_hallim:BEDToolsFlexible2010}. For all features we considered the 20kb upstream and downstream of each IR respectively. For each histone modification, we computed the mean ChIP signal around the IRs. For ATAC-seq, we calculated the total number of peaks with the bedtools map count option. To look for TF motifs we counted the numbers of each motif for TFs expressed in K562s (see above) in each surrounding IR sequence using FIMO \cite{grantce_noblews:FIMOScanning2011} (MEME suite 5.0.4). Because this resulted in a long list of TFs we further filtered the TFs to include only those with a significant correlation with MIN levels in the regression model. To determine the numbers of enhancers interacting with each IR we annotated the loops called from peakHiC above with chromHMM enhancer annotations using the GenomicInteractions R package \cite{harmstonn_lenhardb:GenomicInteractionsBioconductor2015} and counted the number of enhancers. 

For the extrinsic features, we calculated the proportion of cells in the \enquote{stem-like} substate and \enquote{differentiated} substate and different cell cycle phases based on the barcodes that appeared in those substates. We removed IR locations that have less than 30 cells in any of the substates.  

We used the glm function in R (version 3.6.3) to fit logistic regression models. We separated the IR locations into top 20\% MIN and bottom 20\% MIN and used logistic regression to classify locations. We first fit a model with just local sequence features (chromatin modifications, number of TF motifs, number of loops, whether the IR location is in a gene, GC content and the number of ATAC-seq peaks). We used data from one experiment for training the model and used data from another experiment as a holdout set of data to estimate the performance of the classifier. We next fit a model with cellular information for each IR location: proportion of cells with data for the IR location in S phase of the cell cycle, in G2 phase and the proportion of cells that are in the \enquote{stem-like} substate of K562 cells \cite{litzenburgerum_changhy:SinglecellEpigenomic2017}. Lastly, we fit a model that incorporated the extrinsic features and the significant predictors from the intrinsic features model.

\subsection*{Transcriptome analyses associated with SARGENT}

\subsection*{Processing the single-cell transcriptome data}

The single-cell RNAseq data was processed with CellRanger 6.0.1 and SCANPY 1.9.1 \cite{wolffa_theisfj:SCANPYLargescale2018}. Briefly, the raw reads were processed with the standard single-cell expression cell line pipeline line. The resulting expression matrix was then imported into SCANPY for further visualization and clustering. 

\subsection*{Identifying single cell clones}

We identified the individual clones for Pool 4 which contained cells that grew out of 100 two-cell clones. Since most of the clones will have unique integrations into unique genomic locations, the cells that grew out from the same clone will have identical unique sets of locBCs. Due to the dropout rates associated with scRNAseq methods, not all barcodes will be present in all cells, nor will the cell barcodes be uniquely linked to correct sets of locBCs. To identify the barcodes belonging to the same clone, we first recorded locBCs that are linked by a given cell barcode. We then filtered the locBC list associated with a given cellBC based on the number of UMIs associated with these locBC. At this step, we used a knee point detection algorithm \cite{satopaav_raghavanb:FindingKneedle2011} that automatically detects the inflection point of the ordered UMI counts histogram. After filtering for locBCs that appear in more than 5 cells, we constructed a clonal graph by linking locBCs that co-occur in the same cells.

\subsection*{Validation of individual clones}

We extracted gDNA from 16 clones that were grown out from Pool 4. We then amplified the barcodes from each clone using Q5 High Fidelity 2X Master Mix (NEB \#M0492) with primers specific to our reporter gene (CAS P58-59). For each clone, we performed 4 PCRs and pooled the PCRs for purification. 4ng from each clone was then further amplified with 2 rounds of PCR to add Illumina sequencing adapters (CAS P60-63). The barcodes were sequenced on the Illumina NextSeq platform. 

\subsection*{Estimating intrinsic vs extrinsic noise}

To understand how cellular environments affect IR expression, we computed the mean and standard deviation from all IR locations in the same cell. Since standard deviation is expected to increase with mean, we calculated the standard deviation/mean for each cell, which we termed the fluctuation index. To establish the null distributions, we randomly shuffled the cell labels for each clone and computed fluctuation indices for the shuffled cells. 

Intrinsic and extrinsic noise were estimated using the statistical framework developed for the dual-reporter experiment \cite{fuaq_pachterl:EstimatingIntrinsic2016}. In our experiment, single-cell expression differences among IR locations are treated as the intrinsic portion of the noise. We first extracted the pairwise expression level for IR locations in every single cell, then applied the above framework. The derivation is abbreviated and can be found in the original publication. Briefly, let C denote the expression for the first locBC in the cell, Y denote the expression for the second locBC in the cell and n denote the number of cells. 

Let $\eta_{\text {ext}}$ denote the extrinsic noise, and it can be calculated as:
$$
\eta_{\text {ext }}=\frac{1}{a \bar{C} \bar{Y}}\left(\sum_{i=1}^n C_i Y_i-n \bar{C} \bar{Y}\right)
$$
\centerline{where}
$$
% \begin{aligned} 
a=(n-1)\left(1+\frac{1}{n}\right)+\frac{1}{\rho^2} \\
$$
$$
\rho=\frac{\operatorname{Cov}[C, Y]}{\sqrt{\operatorname{Var}[C]} \sqrt{\operatorname{Var}[Y]}}
% \end{aligned}
$$
\\

Similarly, let $\eta_{\text {int}}$ denote the intrinsic noise, and it can be calculated as:
$$
% \begin{aligned}
\eta_{\text {int }}=\frac{1}{2 a \bar{C} \bar{Y}}\left(\sum_{i=1}^n\left(C_i-Y_i\right)^2-n(\bar{C}-\bar{Y})^2\right) 
% \end{aligned}
$$
\centerline{where}
$$
% \begin{aligned}
a=\frac{2 n^3-7 n+6}{2\left(n^2-n\right)}+\frac{2-n}{n^2-n} \frac{\rho}{1-\rho}+\frac{1}{2\left(n^2-n\right)}\left(\frac{\rho}{1-\rho}\right)^2 \\
$$
$$
\rho=\frac{\operatorname{Cov}[C, Y]}{\sqrt{\operatorname{Var}[C]} \sqrt{\operatorname{Var}[Y]}}
% \end{aligned}
$$

\subsection*{Cell substate impact on expression mean and noise}

To compute cell substate specific expression mean and noise at different genomic locations, individual cells were assigned a cell cycle phase of G1, S, or G2/M using a previously reported set of cell-cycle specific marker genes with SCANPY 1.9.1 \cite{wolffa_theisfj:SCANPYLargescale2018}. For the stem-like substate analysis, we clustered cells based on their transcriptomes and assigned cells in the CD24 high cluster as CD24+ cells \cite{litzenburgerum_changhy:SinglecellEpigenomic2017}. To ensure an accurate measurement of expression mean and noise, genomic locations with less than 15 cells in any phase were excluded from the cell cycle analysis. Based on this filtering criterion, 345 out of 939 genomic locations were used for this analysis. To determine the impact of cellular substates on gene expression noise, we calculated the proportion of cells in different cellular substates for each clone. For each clone, we also calculated the average mean and variance of all the IRs in that clone. 

\subsection*{Transgene integration analysis}

To examine whether the integration of a trans-gene alters endogenous gene expression, we first identified IR locations that were integrated into a gene body. Since the IR insertion only occurs in a single clone, we computed pseudobulk expression from cells in the clone using decouplerR 1.1.0 \cite{badia-i-mompelp_saez-rodriguezj:DecoupleREnsemble2022}. We then randomly sampled the same number of cells from all the other clones and used the pseudobulk expression from these cells as wild-type expression. To determine whether the expression in the IR clone is significantly different from wild-type expression, we computed the \textit{p}-value of differential expression using a Fisher’s exact test. 

\section{Supplementary Figures}

\beginsupplement

\begin{suppfigure}[htbp]  
    \centering
    \phantomlabel{fig:cas_figureS1a} 
    \phantomlabel{fig:cas_figureS1b}
    \phantomlabel{fig:cas_figureS1c}
    \phantomlabel{fig:cas_figureS1d}
    \includegraphics[width=\linewidth]{figures/cas_figureS1.ai}
    \caption[smFISH corroborates with SARGENT measurements.]{
        \textbf{smFISH corroborates with SARGENT measurements. (A,B)}
        Mean and noise levels of two IR locations (loc1 \subref{fig:cas_figureS1a} and loc2 \subref{fig:cas_figureS1b}) measured by SARGENT. Values were normalized (Z-scored) for comparison across different experiments.
        \textbf{(C,D)}
        Mean and noise levels of the same two IR locations (loc1 \subref{fig:cas_figureS1c} and loc2 \subref{fig:cas_figureS1d}) measured with smFISH. Error bars represent one std from two biological replicates. 
    }
    \label{fig:cas_figureS1}
\end{suppfigure}

\begin{suppfigure}[htbp]  
    \centering
    \phantomlabel{fig:cas_figureS2a} 
    \phantomlabel{fig:cas_figureS2b}
    \phantomlabel{fig:cas_figureS2c}
    \phantomlabel{fig:cas_figureS2d}
    \includegraphics[width=\linewidth]{figures/cas_figureS2.ai}
    \caption[Measurements of mean-independent noise across different chromosomal environments.]{
        \textbf{Measurements of mean-independent noise (MIN) across different chromosomal environments.}
        \subref{fig:cas_figureS2a}
        IR locations are distributed all throughout the genome. Each black bar above the ideogram represents a separate integration.
        \subref{fig:cas_figureS2b}
        IR locations are distributed across different chromatin types.
        \textbf{(C,D)}
        Expression mean is well correlated with fano factor \subref{fig:cas_figureS2c} and CV\textsuperscript{2} \subref{fig:cas_figureS2d}.  
    }
    \label{fig:cas_figureS2}
\end{suppfigure}

\begin{suppfigure}[htbp]  
    \centering
    \includegraphics[width=\linewidth]{figures/cas_figureS3.ai}
    \caption[Model results for high mean vs low mean.]{
        \textbf{Model results for high mean vs low mean.}
        Weights of features from the gene expression mean model using only intrinsic genomic features. Red bars: \textit{p}-value $<$ 0.05; Pink bars: 0.05 $\leq$ \textit{p}-value $<$ 0.1 from the logistic regression model.  
    }
    \label{fig:cas_figureS3}
\end{suppfigure}

\begin{suppfigure}[htbp]  
    \centering
    \phantomlabel{fig:cas_figureS4a} 
    \phantomlabel{fig:cas_figureS4b}
    \phantomlabel{fig:cas_figureS4c}
    \includegraphics[width=\linewidth]{figures/cas_figureS4.ai}
    \caption[Clonal identification allows for separation of intrinsic and extrinsic noise.]{
        \textbf{Clonal identification allows for separation of intrinsic and extrinsic noise.}
        \subref{fig:cas_figureS4a}
        Histogram of number of integrations per clone.
        \subref{fig:cas_figureS4b}
        Mock histogram showing the expected distributions if noise was either all intrinsic or all extrinsic.
        \subref{fig:cas_figureS4c}
        Histogram of measured fluctuation indices (defined as the standard deviation/mean of all IRs in a cell) for 10 random clones. The shuffled distribution represents the distribution after cell labels have been shuffled, which simulates the case when all noise is intrinsic.   
    }
    \label{fig:cas_figureS4}
\end{suppfigure}

\begin{suppfigure}[htbp]  
    \centering
    \phantomlabel{fig:cas_figureS5a} 
    \phantomlabel{fig:cas_figureS5b}
    \phantomlabel{fig:cas_figureS5c}
    \phantomlabel{fig:cas_figureS5d}
    \phantomlabel{fig:cas_figureS5e}
    \phantomlabel{fig:cas_figureS5f}
    \includegraphics[width=\linewidth]{figures/cas_figureS5.ai}
    \caption[Cell cycle and CD24 states partially explain extrinsic noise.]{
        \textbf{Cell cycle and CD24 states partially explain extrinsic noise.}
        \subref{fig:cas_figureS5a}
        UMAP clustering reveals CD24+ cell population.
        \subref{fig:cas_figureS5b}
        CD24+ cells express stemness and proliferation marker gene GATA1.
        \textbf{(C,D)}
        Violin plots of MIN \subref{fig:cas_figureS5c} and mean \subref{fig:cas_figureS5d} levels in different phases of the cell cycle. \textit{P}-values were calculated using the Mann-Whitney-Wilcoxon test. Legend: $\ast$: 0.01 $<$ \textit{p}-value $\leq$ 0.05, $\ast\ast$: 0.001 $<$ \textit{p}-value $\leq$ 0.01, $\ast\ast\ast$: 0.0001 $<$ \textit{p}-value $\leq$ 0.001, $\ast\ast\ast\ast$: \textit{p}-value $\leq$ 0.0001. ns: not significant.
        \subref{fig:cas_figureS5e}
        Gene expression mean model improves after the addition of extrinsic features.
        \subref{fig:cas_figureS5f}
        Weights of features from the mean model using both intrinsic genomic and extrinsic features. Red bars: \textit{p}-value $<$ 0.05; Pink bars: 0.05 $\leq$ \textit{p}-value $<$ 0.1 from the logistic regression model. 
    }
    \label{fig:cas_figureS5}
\end{suppfigure}

\begin{suppfigure}[htbp]  
    \centering
    \phantomlabel{fig:cas_figureS6a} 
    \phantomlabel{fig:cas_figureS6b}
    \includegraphics[width=\linewidth]{figures/cas_figureS6.ai}
    \caption[IR integrations have little impact on endogenous expression.]{
        \textbf{IR integrations have little impact on endogenous expression.}
        \subref{fig:cas_figureS4a}
        Bar plot number of IRs in endogenous genes per clone.
        \subref{fig:cas_figureS4b}
        Shuffling IR-endogenous gene labels results in no differentially expressed genes.   
    }
    \label{fig:cas_figureS6}
\end{suppfigure}

\clearpage

\section{Supplementary Tables}

\begin{supptable}[htb]
    \centering
    \caption{Primers used in this study}
      \begin{tabular}{lp{13cm}}
      \toprule
      \textbf{Name} & \textbf{Sequence} \\
      \midrule
      CAS P20 & CTACACGACGCTCTTCCGATCT \\
      CAS P32 & GTCAGATGTGTATAAGAGACAG \\
      CAS P41 & GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCTTCTCGCTTCGAGTCTAGA \\
      CAS P42 & GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCTAGCTCTCGCTTCGAGTCTAGA \\
      CAS P45 & TCTCGCTTCGAGTCTAGA \\
      CAS P48 & CAAGCAGAAGACGGCATACGAGATNNNNNNNNNGTGACTGGAGTTCAGAC \\
      CAS P55 & GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCTTAGGACCGGCCTTAAAGC \\
      CAS P57 & GGGGCACAAGCTTAATTGAGAGCTCTCGCTTCGAGTCTAGANNNNNNNNNNNNNNNNCTCGAGTTGTGGCCGGCCCTT \\
      CAS P58 & TGCCTGGCGTCTACTATGTG \\
      CAS P59 & AACGCCAGGGTTTTCCCAA \\
      CAS P60 & CTTTCCCTACACGACGCTCTTCCGATCT(N[1-4])CTCGCTTCGAGTCTAGA \\
      CAS P61 & GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCTGCCAGGGTTTTCCCAAC \\
      CAS P62 & AATGATACGGCGACCACCGAGATCTACACNNNNNNACACTCTTTCCCTACACGACGCT \\
      CAS P63 & CAAGCAGAAGACGGCATACGAGATNNNNNNNNNGTGACTGGAGTTCAGACGTG \\
      CAS P64 & GTACGCATGCCGCATGATTATCTTTAACGTACGTCAC \\
      CAS P65 & ACGACGCTCTTCCGATCTGCTCGA(N[1-4])GTACGTCACAATATGATTATCTTTCTAG \\
      CAS PP2 & ATCTACACTCTTTCCCTACACGACGCTCTTC \\
      CAS PP4 & AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACA \\
      CAS CS2 & CGAGATCTACACTCGTCGGCAGCGTCAGATGTGTATAAGAGACAG \\
      CAS CS4 & AATGATACGGCGACCACCGAGATCTACACTCGTCG \\
      \bottomrule
      \end{tabular}%
    \label{tab:cas_tableS1}%
  \end{supptable}%
  
