\chapter{Discussion}
\label{chap:conclusion}
\tightlists

Work in the past few decades has shown that cell-to-cell variability in gene expression is ubiquitous. How does an organism function reliably in the midst of this variability? How does a cell still respond to external and internal signals reliably? How are biological processes regulated with high fidelity in the face of such variability? The work presented in this thesis provides a few more hints about the impact and sources of cell-to-cell variability that sheds a little more light on the answers to these questions.

In the second chapter of this thesis I looked at variability amongst cells in response to the stimulation of the hedgehog signaling pathway. I found that some cells are able to respond faster to this pathway stimulation. Further, I found that overexpression of a single transcription factor was able to make most of the cells respond fast. The cell-to-cell variability in gene expression of this single transcription factor could in principle underlie important developmental decisions. It thus becomes important to understand what underlies the cell-to-cell variability in gene expression of different genes in the genome.

In the third chapter of my thesis I helped develop a new technology to measure the effect of genomic location and cell state on cell-to-cell variability by using a same reporter gene inserted in multiple genomic locations. By looking at how the cell-to-cell variability changes with various sequence features at the different genomic locations I was able to understand what are the genomic features that distinguish genomic regions with high cell-to-cell variability from genomic regions with low cell-to-cell variability. I also found that most of the cell-to-cell variability is determined by the mean level of expression, and removing the effect of the mean on the variance leaves a very small amount of residual variability that we examined.

In this chapter, I will discuss the implications of these results. Specifically, I will first touch on what are the gaps in the field of cell-to-cell variability that this study addresses. Next, I will discuss how I think the cell makes decisions in the face of so much cell-to-cell variability. Then, I will discuss future directions of this work and finally, I will examine if and when variability in gene expression matters.

\subsection{What are the gaps that the work in this thesis filled?}

We started the project in Chapter 2 by asking if we could explain the cell-to-cell variability seen in single-cell RNA sequencing. Whenever studies perform single-cell RNA sequencing and visualize the cells on a dimensional reduction plot such as UMAP, they see a huge blob for each cell-type. We started by asking what generates this blob, is it all just technical noise, or could some of it be biological variability. We then hypothesized that different transcription factors could be active in different parts of the UMAP, and turning on different target genes, we validated this hypothesis in Chapter 2 for one subset of the UMAP. We also wanted to see if some cells turn on the hedgehog pathway stochastically in untreated 3T3 cells. We didn't find cells that turned on the pathway stochastically, this result indicates the robustness of the signaling pathway to random fluctuations in the cell. Our final question for this project turned into what makes some cells respond faster to the hedgehog stimulation compared to others, we found one transcription factor that has the ability to do this. While I have studied variability in one signaling pathway in detail in Chapter 2 similar approaches using single-cell RNA sequencing and trajectory analysis can be used to dissect any biological process and understand cell-to-cell variability in that process.

The work in Chapter 3 started out as a conversation between three graduate students about how we could extend bulk approaches that look at the effect of chromatin on gene expression to the single-cell level. We extended the innovations of Akhtar et al. \cite{akhtarw_vansteenselb:ChromatinPosition2013} and figured out a way to increase the efficiency of capture of our reporter transcripts in single-cell RNA sequencing. We designed a reporter sequence with a specific capture sequence at the 3' end which would bind to complementary sequences on the beads from 10X Genomics in addition to the polyA binding to the polyT sequences. We simultaneously measured the transcriptome of the cells which provided a readout of the trans-environment for our reporter gene and allowed us to estimate the effect of cell-state on noise. To our knowledge this is the first study that measured the impact of the measured transcriptional cell-state on gene expression noise. We also measured the gene expression noise of a reporter at close to a 1000 locations and identified features that separate out high noise locations from low noise locations. Our approach can be extended to cover an order of magnitude higher locations, this increase in the number of locations combined with further annotation of the genome will likely uncover more features that explain the noise across the genome with a high accuracy. I will discuss more about this approach in the next section.

\section{How are cellular decisions made in the face of noise?}

In the first project I showed that the cell-to-cell variability of a single transcription factor \emph{Prrx1} can influence whether a cell responds to a perturbation or not. One model that might explain how the cell responds is by using a threshold model for any cellular decision. The threshold model suggests that cells that have a level of protein above a certain level are the ones that are poised to respond. Cells that are below a protein level are the ones that don't respond. When a population of cells are induced to express a certain gene, the mean level of expression increases but often the mean level increase is accompanied by an increase in the cell-to-cell variability. However, the proportion of cells that respond to a stimuli is determined by the proportion of cells that are able to express the gene above a certain level. This model fits with what I observe in \emph{Prrx1}.

In the model for hedgehog signaling in \label{Chap:hedgehog} we hypothesized that stochastic activity of transcription factors might underlie why some cells are able to respond faster to hedgehog perturbation vs others. We identified a lost of candidate transcription factors that are overexpressed in the fast responding cells and tested five TFs. We found that overexpressing one of the transcription factors Prrx1 is able to increase the proportion of cells that respond faster. In a threshold model of cellular decision making, cells that have sufficient activity of Prrx1, above a certain threshold, are able to respond to the hedgehog stimulation.Â  When we induce Prrx1 expression, we increase the mean and variance of Prrx1. However the proportion of cells that express Prrx1 above a certain threshold are able to respond. A follow-up question from this study is how many such regulators exist and how might we find them?

We identified the fast-responder gene signature consisting of 300 candidates. In theory, there might be some false positives amongst this list of 300 genes, these are just a set of candidate genes. However, in our model a TF that is able to express a sufficient fraction of these 300 genes is able to make the cells respond. When we overexpress Prrx1, we showed that we misexpress 74 of the 300 genes. Could the 74 genes be the ones that matter? I think this question still remains to be answered. It could be even that only a small fraction of the 74 genes matters. Or it could be the case that Prrx1 operates through a different mechanism where it turns on Gli2, and that's all that matters for faster hedgehog signaling. We tried approaches looking for motifs in the 300 genes, and identifying enriched TFs in the promoter regions of the 300 genes, however we were limited by the number of TFs that we could test experimentally for their effect on the hedgehog response. 

\subsection{Future Directions}

In this section I will describe future directions of the two projects that I worked on. I hope this section will prove somewhat useful to anyone considering studying cell-to-cell variability in gene expression. 

In the second chapter, I was able to show how hedgehog signaling was faster in cells when I overexpressed a single transcription factor Prrx1. However, it remains unknown how many other transcription factors are able to do this. It would be helpful if we could computationally predict other similar genes that reduce this variability from the list of candidate transcription factors instead of testing them out one by one experimentally. Indeed, we tried using computational approaches like cell-oracle \cite{kamimoto2020} to predict which TF perturbation might change the cell state from the slow-responder to fast-responder cell state. These approaches which integrate multiple lines of evidence including cell-state differential expression, transcription factor binding data, and chromatin accessibility data to predict the effects of TF perturbations. However, such approaches are still in their infancy and there is a lot of work that can be done here to develop computational approaches to predict perturbation effects. Data such as what I have generated for this thesis work - RNA-seq data after overexpressing multiple transcription factors either through plasmid transfection or in stably integrated cell-lines can be used to train such computational models. If these models are accurate enough, this would could potentially minimize the number of functional experiments and provide a list of high confidence genes to test.

On the technology side of things, in the second and third chapters we  used single-cell RNA sequencing to look at the cell-to-cell variability of different genes. Single-cell RNA sequencing, as discussed in Chapter 1, in-theory helps to measure the expression of all polyA transcripts within a cell. However, the current methods of transcript capture are still far from perfect. Only a small fraction (5 - 20\%) of the overall transcripts within a cell are captured \cite{battich_control_2015} and it is unclear how random this dropout is amongst transcripts of various genes. I am optimistic that these technical challenges will be mostly solved over the next decade, this would further increase the statistical power of the approaches used in this thesis to detect true effects. The fold-changes that we detect in the expression of transcription factors between the slow responders and fast-responders is pretty small. Part of this small fold-change could be drop-out in either population resulting in false-positives in the DE genes. More precise measurements of transcript abundances in single-cells could potentially result in a smaller and more high-confidence set of candidate genes that are differentially expressed between sub-states. In Chapter 3, the mean independent noise is a small fraction of the overall noise levels of a gene. Having more precise measurements of noise will also improve our ability to quantify this noise accurately. Approaches such as MERFISH \cite{xia2019pnasu} which use imaging to measure multiple transcripts within a cell could provide an alternative avenue to sequencing. However multiplexed imaging approaches are currently still challenging to setup, and harder to scale compared to sequencing. This may very well change in the future.

We've used a computational method for tracking the trajectory that cells take during hedgehog signaling. This trajectory analysis approach can be sensitive to parameter changes, and in the future I expect there may be better technologies to trace the series of cell-state changes that a cell undergoes when confronted with a perturbation. Already barcode-based approaches are available \cite{Biddy2018-ct} \cite{moudgil2020c}, however even these approaches are not fool-proof and can be cost-prohibitive due to barcode dropout issues. The availability of better lineage tracing technologies will further enable the study of cell-to-cell variability and shed light on why approaches like reprogramming have low efficiency despite starting with a uniform pool of cells.

In Chapter 2, we've integrated the same reporter gene in multiple locations to understand what features are associated with cell-to-cell variability. In the future, one might imagine integrating different reporter genes or the same gene with different cis-regulatory elements genome-wide and study cell-to-cell variability in a similar approach. Indeed, similar approaches have already been used to study the impact of genomic location on mean expression \cite{hong2022gr}, however precise single-cell measurements will extend this to studying the variance in gene expression. This approach would be valuable for gene-therapy approaches. For example, if there are interactions between a gene and it's local environment such experiments will enable choosing locations with low variability. The same cis-regulatory element might have the same mean at different locations but different cell-to-cell variability and current bulk approaches that study just the mean level of expression are blind to such phenomena.

Finally, I've used 10X Genomics which is a droplet based method for single-cell RNA sequencing for the experiments in Chapters 2 and 3. This let me measure around 10,000 single cells per sample in a single experiment. However in Chapter 3 we need a lot more cells to get a precise estimate of the variance. This would also enable models with better accuracy to identify significant co-variates that affect noise since we could use the noise as a quantitative phenotype in a regression model instead of coarsely binning locations as high noise vs low noise locations. Indeed, the next generation of single-cell RNA-sequencing approaches are already available \cite{caoj_shendurej:ComprehensiveSinglecell2017} which are able to scale the experiments to hundreds of thousands of cells. It will likely be a relatively straightforward approach to scale up the studies in this work using the next generation of single-cell RNA sequencing assays. 

\subsection{Does cell-to-cell variability matter?}

Through the course of graduate school, I often heard statements that took some form of "Biology is messy". While the same statement was used to describe different classes of biological phenomena, there is one class of phenomenon that is relevant to the work described in this thesis. Cells or even individuals with the same genome don't always behave the same way when exposed to identical stimuli. A part of this phenotypic variability is the result of the stochastic nature of gene expression that I have sought to understand in this thesis.

We are entering an era where genome editing is becoming increasingly common. There are reports out about genome editing of embryos and embryo selection using polygenic risk scores \cite{ma2017n} \cite{pereira2022hr}. At the same time we are sequencing large number of individuals using a genotype first approach and finding that large number of clinically rated pathogenic mutations have incomplete penetrance \cite{forrest2022j}. When these mutations are modeled in the lab, even model organisms show stochastic phenotypes \cite{marmion2023cb}. These early results indicate that we will need to account for variability when genome-editing and gene therapy approaches become more mainstream for different applications. How might we able to deal with variability?

One view about cell-to-cell variability is that the mean expression is all that matters since the mean explains a large part of the variance. However depending on the application the variability might still matter. For example if a gene-therapy needs a molecule, say an enzyme, to be synthesized in the cell at a constant rate then simply specifying the mean expression might not be sufficient. Specifying the burst frequency and the burst size would allow for a more precise control of the therapeutic. Some therapies might prefer a high burst frequency, and perhaps a smaller burst size, over others that prefer less frequent bursts.

While such precise control of biological systems sounds speculative at this point in time, if identification of regulatory elements controlling variability \cite{zhao2023ng} and modeling approaches such as those described in this thesis become increasingly accurate at predicting and estimating noise such applications might not be too far in our collective future. It is indeed an fascinating time to be a geneticist.
